<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="BioCor">
<title>Advanced usage of BioCor • BioCor</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Advanced usage of BioCor">
<meta property="og:description" content="BioCor">
<meta property="og:image" content="man/figures/logo.png">
<meta property="og:image:alt" content="BioCor logo">
<meta name="twitter:card" content="summary">
<meta name="twitter:creator" content="@Lluis_Revilla">
<meta name="twitter:site" content="@Lluis_Revilla">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">BioCor</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="Released version">1.25.1</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/BioCor_1_basics.html">About BioCor</a>
    <a class="dropdown-item" href="../articles/BioCor_2_advanced.html">Advanced usage of BioCor</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/llrs/BioCor/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.svg" class="logo" alt=""><h1>Advanced usage of BioCor</h1>
                        <h4 data-toc-skip class="author">Lluís
Revilla</h4>
            <address class="author_afil">
      August Pi i Sunyer Biomedical Research Institute (IDIBAPS); Liver
Unit, Hospital
Clinic<br><a class="author_email" href="mailto:#"></a><a href="mailto:lrevilla@clinic.cat" class="email">lrevilla@clinic.cat</a>
      </address>
                              <h4 data-toc-skip class="author">Juan José
Lozano</h4>
            <address class="author_afil">
      Centro de Investigación Biomédica en Red de Enfermedades Hepaticas
y Digestivas (CIBEREHD); Barcelona,
Spain<br><h4 data-toc-skip class="author">Pau
Sancho-Bru</h4>
            <address class="author_afil">
      August Pi i Sunyer Biomedical Research Institute (IDIBAPS); Liver
Unit, Hospital Clinic<br><h4 data-toc-skip class="date">27 April 2023</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/llrs/BioCor/blob/HEAD/vignettes/BioCor_2_advanced.Rmd" class="external-link"><code>vignettes/BioCor_2_advanced.Rmd</code></a></small>
      <div class="d-none name"><code>BioCor_2_advanced.Rmd</code></div>
    </address>
</address>
</div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      <p>Describes how to use the BioCor package to answer several
      biological questions and how to use functional similarities with
      other measures.</p>
    </div>
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>In this vignette we assume that the reader is already familiarized
with the <a href="https://bioconductor.org/packages/3.17/BioCor/vignettes/BioCor_1_basics.html" class="external-link">introduction
vignette</a>, but wants to know how can it help to answer other
questions in other situations</p>
<p>We follow the same convention about names of the objects used
<code>genesReact</code> and <code>genesKegg</code> are the list with
information about the pathways the genes are involved in.</p>
</div>
<div class="section level2">
<h2 id="merging">Merging similarities<a class="anchor" aria-label="anchor" href="#merging"></a>
</h2>
<p>If one calculates similarities with KEGG data and Reactome or other
input for the same genes or clusters BioCor provides a couple of
functions to merge them.</p>
<p>We can set a weight to each similarity input with
<code>weighted.sum</code>, multiply them also using a weight for each
similarity (with <code>weighted.prod</code>), doing the mean or just
adding them up. Similarities allow us to apply a function to combine the
matrices of a list. Here we use some of the genes used in the first
vignette:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">kegg</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/mgeneSim.html">mgeneSim</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"672"</span>, <span class="st">"675"</span>, <span class="st">"10"</span><span class="op">)</span>, <span class="va">genesKegg</span><span class="op">)</span></span>
<span><span class="va">react</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/mgeneSim.html">mgeneSim</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"672"</span>, <span class="st">"675"</span>, <span class="st">"10"</span><span class="op">)</span>, <span class="va">genesReact</span><span class="op">)</span></span>
<span><span class="co">## We can sum it adding a weight to each origin</span></span>
<span><span class="fu"><a href="../reference/weighted.html">weighted.sum</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">kegg</span><span class="op">[</span><span class="st">"672"</span>, <span class="st">"675"</span><span class="op">]</span>, <span class="va">react</span><span class="op">[</span><span class="st">"672"</span>, <span class="st">"675"</span><span class="op">]</span><span class="op">)</span>, w <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.3</span>, <span class="fl">0.7</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">## [1] 0.7247289</span></span>
<span></span>
<span><span class="co">## Or if we want to perform for all the matrix</span></span>
<span><span class="co">## A list of matrices to merge</span></span>
<span><span class="va">sim</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="st">"kegg"</span> <span class="op">=</span> <span class="va">kegg</span>, <span class="st">"react"</span> <span class="op">=</span> <span class="va">react</span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/similarities.html">similarities</a></span><span class="op">(</span><span class="va">sim</span>, <span class="va">weighted.sum</span>, w <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.3</span>, <span class="fl">0.7</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">##           672       675        10</span></span>
<span><span class="co">## 672 1.0000000 0.7247289 0.1552549</span></span>
<span><span class="co">## 675 0.7247289 1.0000000 0.1577274</span></span>
<span><span class="co">## 10  0.1552549 0.1577274 1.0000000</span></span>
<span><span class="fu"><a href="../reference/similarities.html">similarities</a></span><span class="op">(</span><span class="va">sim</span>, <span class="va">weighted.prod</span>, w <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.3</span>, <span class="fl">0.7</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">##           672         675          10</span></span>
<span><span class="co">## 672 0.2100000 0.017310195 0.000000000</span></span>
<span><span class="co">## 675 0.0173102 0.210000000 0.000383872</span></span>
<span><span class="co">## 10  0.0000000 0.000383872 0.210000000</span></span>
<span><span class="fu"><a href="../reference/similarities.html">similarities</a></span><span class="op">(</span><span class="va">sim</span>, <span class="va">prod</span><span class="op">)</span></span>
<span><span class="co">##           672         675          10</span></span>
<span><span class="co">## 672 1.0000000 0.082429501 0.000000000</span></span>
<span><span class="co">## 675 0.0824295 1.000000000 0.001827962</span></span>
<span><span class="co">## 10  0.0000000 0.001827962 1.000000000</span></span>
<span><span class="fu"><a href="../reference/similarities.html">similarities</a></span><span class="op">(</span><span class="va">sim</span>, <span class="va">mean</span><span class="op">)</span></span>
<span><span class="co">##           672       675        10</span></span>
<span><span class="co">## 672 1.0000000 0.5412148 0.1108964</span></span>
<span><span class="co">## 675 0.5412148 1.0000000 0.1150172</span></span>
<span><span class="co">## 10  0.1108964 0.1150172 1.0000000</span></span></code></pre></div>
<p>This functions are similar to <code>weighted.mean</code>, except that
first the multiplication by the weights is done and then the
<code>NA</code>s are removed:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/stats/weighted.mean.html" class="external-link">weighted.mean</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="cn">NA</span><span class="op">)</span>, w <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.5</span>, <span class="fl">0.5</span><span class="op">)</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">## [1] 1</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/stats/weighted.mean.html" class="external-link">weighted.mean</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">0.5</span>, <span class="cn">NA</span><span class="op">)</span>, w <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.5</span>, <span class="fl">0.25</span>, <span class="fl">0.25</span><span class="op">)</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">## [1] 0.8333333</span></span>
<span><span class="fu"><a href="../reference/weighted.html">weighted.sum</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="cn">NA</span><span class="op">)</span>, w <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.5</span>, <span class="fl">0.5</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">## [1] 0.5</span></span>
<span><span class="fu"><a href="../reference/weighted.html">weighted.sum</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">0.5</span>, <span class="cn">NA</span><span class="op">)</span>, w <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.5</span>, <span class="fl">0.25</span>, <span class="fl">0.25</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">## [1] 0.625</span></span>
<span><span class="fu"><a href="../reference/weighted.html">weighted.prod</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="cn">NA</span><span class="op">)</span>, w <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.5</span>, <span class="fl">0.5</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">## [1] 0.5</span></span>
<span><span class="fu"><a href="../reference/weighted.html">weighted.prod</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">0.5</span>, <span class="cn">NA</span><span class="op">)</span>, w <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.5</span>, <span class="fl">0.25</span>, <span class="fl">0.25</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">## [1] 0.0625</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="assessing-a-differential-study">Assessing a differential study<a class="anchor" aria-label="anchor" href="#assessing-a-differential-study"></a>
</h2>
<p>In this example we will use the functional similarities in a
classical differential study.</p>
<div class="section level3">
<h3 id="obtaining-data">Obtaining data<a class="anchor" aria-label="anchor" href="#obtaining-data"></a>
</h3>
<p>We start using data from the <a href="https://bioconductor.org/help/workflows/rnaseqGene/#differential-expression-analysis" class="external-link">RNAseq
workflow</a> and following the analysis comparing treated and
untreated:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/message.html" class="external-link">suppressPackageStartupMessages</a></span><span class="op">(</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st">"airway"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"airway"</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://github.com/mikelove/DESeq2" class="external-link">"DESeq2"</a></span><span class="op">)</span></span>
<span></span>
<span><span class="va">dds</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/DESeqDataSet.html" class="external-link">DESeqDataSet</a></span><span class="op">(</span><span class="va">airway</span>, design <span class="op">=</span> <span class="op">~</span> <span class="va">cell</span> <span class="op">+</span> <span class="va">dex</span><span class="op">)</span></span>
<span><span class="va">dds</span><span class="op">$</span><span class="va">dex</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/relevel.html" class="external-link">relevel</a></span><span class="op">(</span><span class="va">dds</span><span class="op">$</span><span class="va">dex</span>, <span class="st">"untrt"</span><span class="op">)</span></span>
<span><span class="va">dds</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/DESeq.html" class="external-link">DESeq</a></span><span class="op">(</span><span class="va">dds</span><span class="op">)</span></span>
<span><span class="co">## estimating size factors</span></span>
<span><span class="co">## estimating dispersions</span></span>
<span><span class="co">## gene-wise dispersion estimates</span></span>
<span><span class="co">## mean-dispersion relationship</span></span>
<span><span class="co">## final dispersion estimates</span></span>
<span><span class="co">## fitting model and testing</span></span>
<span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/DESeq2/man/results.html" class="external-link">results</a></span><span class="op">(</span><span class="va">dds</span>, alpha <span class="op">=</span> <span class="fl">0.05</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## out of 33469 with nonzero total read count</span></span>
<span><span class="co">## adjusted p-value &lt; 0.05</span></span>
<span><span class="co">## LFC &gt; 0 (up)       : 2211, 6.6%</span></span>
<span><span class="co">## LFC &lt; 0 (down)     : 1817, 5.4%</span></span>
<span><span class="co">## outliers [1]       : 0, 0%</span></span>
<span><span class="co">## low counts [2]     : 16687, 50%</span></span>
<span><span class="co">## (mean count &lt; 7)</span></span>
<span><span class="co">## [1] see 'cooksCutoff' argument of ?results</span></span>
<span><span class="co">## [2] see 'independentFiltering' argument of ?results</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">res</span><span class="op">$</span><span class="va">log2FoldChange</span>, <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="va">res</span><span class="op">$</span><span class="va">padj</span><span class="op">)</span>,</span>
<span>    pch <span class="op">=</span> <span class="fl">16</span>, xlab <span class="op">=</span> <span class="st">"log2FC"</span>,</span>
<span>    ylab <span class="op">=</span> <span class="st">"-log10(p.ajd)"</span>, main <span class="op">=</span> <span class="st">"Untreated vs treated"</span></span>
<span><span class="op">)</span></span>
<span><span class="va">logFC</span> <span class="op">&lt;-</span> <span class="fl">2.5</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/abline.html" class="external-link">abline</a></span><span class="op">(</span>v <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="op">-</span><span class="va">logFC</span>, <span class="va">logFC</span><span class="op">)</span>, h <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="fl">0.05</span><span class="op">)</span>, col <span class="op">=</span> <span class="st">"red"</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="BioCor_2_advanced_files/figure-html/simulate-1.png" alt="Volcano plot (log2FC on the horitzonal axis and log(p-value) on the vertical axis) of the airway dataset." width="700"><p class="caption">
Volcano plot. The airway data
</p>
</div>
<p>As we can see here there are around 4000 genes differentially
expressed genes, some of them differentially expressed above 2^2.5.</p>
</div>
<div class="section level3">
<h3 id="selecting-differentially-expressed-genes">Selecting differentially expressed genes<a class="anchor" aria-label="anchor" href="#selecting-differentially-expressed-genes"></a>
</h3>
<p>Usually in such a study one selects genes above certain logFC or fold
change threshold, here we use the absolute value of 2.5:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">fc</span> <span class="op">&lt;-</span> <span class="va">res</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/MathFun.html" class="external-link">abs</a></span><span class="op">(</span><span class="va">res</span><span class="op">$</span><span class="va">log2FoldChange</span><span class="op">)</span> <span class="op">&gt;=</span> <span class="va">logFC</span> <span class="op">&amp;</span> <span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">res</span><span class="op">$</span><span class="va">padj</span><span class="op">)</span>, <span class="op">]</span></span>
<span><span class="va">fc</span> <span class="op">&lt;-</span> <span class="va">fc</span><span class="op">[</span><span class="va">fc</span><span class="op">$</span><span class="va">padj</span> <span class="op">&lt;</span> <span class="fl">0.05</span>, <span class="op">]</span></span>
<span><span class="co"># Convert Ids (used later)</span></span>
<span><span class="va">genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/AnnotationDb-class.html" class="external-link">select</a></span><span class="op">(</span><span class="va">org.Hs.eg.db</span>,</span>
<span>    keys <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span>, keytype <span class="op">=</span> <span class="st">"ENSEMBL"</span>,</span>
<span>    column <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ENTREZID"</span>, <span class="st">"SYMBOL"</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span><span class="co">## 'select()' returned 1:many mapping between keys and columns</span></span>
<span><span class="va">genesFC</span> <span class="op">&lt;-</span> <span class="va">genes</span><span class="op">[</span><span class="va">genes</span><span class="op">$</span><span class="va">ENSEMBL</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">fc</span><span class="op">)</span>, <span class="op">]</span></span>
<span><span class="va">genesFC</span> <span class="op">&lt;-</span> <span class="va">genesFC</span><span class="op">[</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">genesFC</span><span class="op">$</span><span class="va">ENTREZID</span><span class="op">)</span>, <span class="op">]</span></span>
<span><span class="va">genesSim</span> <span class="op">&lt;-</span> <span class="va">genesFC</span><span class="op">$</span><span class="va">ENTREZID</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">genesSim</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">genesFC</span><span class="op">$</span><span class="va">SYMBOL</span></span>
<span><span class="va">genesSim</span> <span class="op">&lt;-</span> <span class="va">genesSim</span><span class="op">[</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/duplicated.html" class="external-link">duplicated</a></span><span class="op">(</span><span class="va">genesSim</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="co"># Calculate the functional similarity</span></span>
<span><span class="va">sim</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/mgeneSim.html">mgeneSim</a></span><span class="op">(</span>genes <span class="op">=</span> <span class="va">genesSim</span>, info <span class="op">=</span> <span class="va">genesReact</span>, method <span class="op">=</span> <span class="st">"BMA"</span><span class="op">)</span></span>
<span><span class="co">## Warning in mgeneSim(genes = genesSim, info = genesReact, method = "BMA"): Some</span></span>
<span><span class="co">## genes are not in the list provided.</span></span></code></pre></div>
<p>Once the similarities for the selected genes are calculated we can
now visualize the effect of each method:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">nas</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/apply.html" class="external-link">apply</a></span><span class="op">(</span><span class="va">sim</span>, <span class="fl">1</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/all.html" class="external-link">all</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span></span>
<span><span class="va">sim</span> <span class="op">&lt;-</span> <span class="va">sim</span><span class="op">[</span><span class="op">!</span><span class="va">nas</span>, <span class="op">!</span><span class="va">nas</span><span class="op">]</span></span>
<span></span>
<span><span class="va">MDSs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/cmdscale.html" class="external-link">cmdscale</a></span><span class="op">(</span><span class="fl">1</span> <span class="op">-</span> <span class="va">sim</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">MDSs</span>, type <span class="op">=</span> <span class="st">"n"</span>, main <span class="op">=</span> <span class="st">"BMA similarity"</span>, xlab <span class="op">=</span> <span class="st">"MDS1"</span>, ylab <span class="op">=</span> <span class="st">"MDS2"</span><span class="op">)</span></span>
<span><span class="va">up</span> <span class="op">&lt;-</span> <span class="va">genes</span><span class="op">[</span><span class="va">genes</span><span class="op">$</span><span class="va">ENSEMBL</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">fc</span><span class="op">)</span><span class="op">[</span><span class="va">fc</span><span class="op">$</span><span class="va">log2FoldChange</span> <span class="op">&gt;=</span> <span class="va">logFC</span><span class="op">]</span>, <span class="st">"SYMBOL"</span><span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/text.html" class="external-link">text</a></span><span class="op">(</span><span class="va">MDSs</span>, labels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">MDSs</span><span class="op">)</span>, col <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/ifelse.html" class="external-link">ifelse</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">MDSs</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="va">up</span>, <span class="st">"black"</span>, <span class="st">"red"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/abline.html" class="external-link">abline</a></span><span class="op">(</span>h <span class="op">=</span> <span class="fl">0</span>, v <span class="op">=</span> <span class="fl">0</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/legend.html" class="external-link">legend</a></span><span class="op">(</span><span class="st">"top"</span>, legend <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Up-regulated"</span>, <span class="st">"Down-regulated"</span><span class="op">)</span>, fill <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"black"</span>, <span class="st">"red"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="BioCor_2_advanced_files/figure-html/pval1-1.png" alt="First two dimensions of a multi dimensional scaling method based on the similarity of genes. Colored if the fold change is above 2.5 (red)." width="1440"><p class="caption">
Functional similarity of genes with logFC above 2,5. Similar genes
cluster together.
</p>
</div>
<p>This plot illustrate that some differentially expressed genes are
quite similar according to their pathways. Suggesting that there might
be a relationship between them. Furthermore, some up-regulated genes
seem functionally related to down-regulated genes indicating a precise
regulation of the pathways where they are involved.</p>
<p>Note that here we are only using 74 genes from the original 127.</p>
</div>
<div class="section level3">
<h3 id="are-differentially-expressed-genes-selected-by-their-functionality">Are differentially expressed genes selected by their
functionality?<a class="anchor" aria-label="anchor" href="#are-differentially-expressed-genes-selected-by-their-functionality"></a>
</h3>
<p>In the previous section we have seen that some differentially
expressed genes are functionally related and that they have a high logFC
value. Are genes differentially expressed more functional related than
those which aren’t differential expressed?</p>
<p>For simplicity we will use a subset of 400 genes represented again in
a volcano plot and we will look for the functional similarities between
those genes:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">250</span><span class="op">)</span></span>
<span><span class="va">subRes</span> <span class="op">&lt;-</span> <span class="va">res</span><span class="op">[</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">res</span><span class="op">$</span><span class="va">log2FoldChange</span><span class="op">)</span>, <span class="op">]</span></span>
<span><span class="va">subs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample.int</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">subRes</span><span class="op">)</span>, size <span class="op">=</span> <span class="fl">400</span><span class="op">)</span></span>
<span><span class="va">subRes</span> <span class="op">&lt;-</span> <span class="va">subRes</span><span class="op">[</span><span class="va">subs</span>, <span class="op">]</span></span>
<span><span class="va">g</span> <span class="op">&lt;-</span> <span class="va">genes</span><span class="op">[</span><span class="va">genes</span><span class="op">$</span><span class="va">ENSEMBL</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">subRes</span><span class="op">)</span>, <span class="st">"ENTREZID"</span><span class="op">]</span></span>
<span><span class="va">gS</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/mgeneSim.html">mgeneSim</a></span><span class="op">(</span><span class="va">g</span><span class="op">[</span><span class="va">g</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">genesReact</span><span class="op">)</span><span class="op">]</span>, <span class="va">genesReact</span>, <span class="st">"BMA"</span><span class="op">)</span></span>
<span><span class="va">deg</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">subRes</span><span class="op">[</span><span class="va">subRes</span><span class="op">$</span><span class="va">padj</span> <span class="op">&lt;</span> <span class="fl">0.05</span> <span class="op">&amp;</span> <span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">subRes</span><span class="op">$</span><span class="va">padj</span><span class="op">)</span>, <span class="op">]</span><span class="op">)</span></span>
<span><span class="va">keep</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">gS</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="va">genes</span><span class="op">[</span><span class="va">genes</span><span class="op">$</span><span class="va">ENSEMBL</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="va">deg</span>, <span class="st">"ENTREZID"</span><span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">subRes</span><span class="op">$</span><span class="va">log2FoldChange</span>, <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="va">subRes</span><span class="op">$</span><span class="va">padj</span><span class="op">)</span>,</span>
<span>    pch <span class="op">=</span> <span class="fl">16</span>, xlab <span class="op">=</span> <span class="st">"log2FC"</span>,</span>
<span>    ylab <span class="op">=</span> <span class="st">"-log10(p.ajd)"</span>, main <span class="op">=</span> <span class="st">"Untreated vs treated"</span></span>
<span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/abline.html" class="external-link">abline</a></span><span class="op">(</span>v <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="op">-</span><span class="va">logFC</span>, <span class="va">logFC</span><span class="op">)</span>, h <span class="op">=</span> <span class="op">-</span><span class="fu"><a href="https://rdrr.io/r/base/Log.html" class="external-link">log10</a></span><span class="op">(</span><span class="fl">0.05</span><span class="op">)</span>, col <span class="op">=</span> <span class="st">"red"</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="BioCor_2_advanced_files/figure-html/setting-1.png" alt="Volcano plot of a subset of 400 genes. " width="700"><p class="caption">
Volcano plot of the subset of 400 genes. This subset will be used in the
following sections
</p>
</div>
<p>We can answer this by testing it empirically:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st">"boot"</span><span class="op">)</span></span>
<span><span class="co"># The mean of genes differentially expressed</span></span>
<span><span class="op">(</span><span class="va">scoreDEG</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/mean.html" class="external-link">mean</a></span><span class="op">(</span><span class="va">gS</span><span class="op">[</span><span class="op">!</span><span class="va">keep</span>, <span class="va">keep</span><span class="op">]</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">## [1] 0.1226829</span></span>
<span><span class="va">b</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/boot/man/boot.html" class="external-link">boot</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">gS</span>, R <span class="op">=</span> <span class="fl">1000</span>, statistic <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span>, <span class="va">i</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="va">g</span> <span class="op">&lt;-</span> <span class="op">!</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">x</span><span class="op">)</span><span class="op">[</span><span class="va">i</span><span class="op">]</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/mean.html" class="external-link">mean</a></span><span class="op">(</span><span class="va">x</span><span class="op">[</span><span class="va">g</span>, <span class="va">i</span><span class="op">]</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span></span>
<span><span class="op">(</span><span class="va">p.val</span> <span class="op">&lt;-</span> <span class="op">(</span><span class="fl">1</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="va">b</span><span class="op">$</span><span class="va">t</span> <span class="op">&gt;</span> <span class="va">scoreDEG</span><span class="op">)</span><span class="op">)</span> <span class="op">/</span> <span class="fl">1001</span><span class="op">)</span></span>
<span><span class="co">## [1] 0.7102897</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/hist.html" class="external-link">hist</a></span><span class="op">(</span><span class="va">b</span><span class="op">$</span><span class="va">t</span>, main <span class="op">=</span> <span class="st">"Distribution of scores"</span>, xlab <span class="op">=</span> <span class="st">"Similarity score"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/abline.html" class="external-link">abline</a></span><span class="op">(</span>v <span class="op">=</span> <span class="va">scoreDEG</span>, col <span class="op">=</span> <span class="st">"red"</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="BioCor_2_advanced_files/figure-html/cluster2-1.png" alt="Histogram of the scores of similarity between several genes. A vertical red line indicates the score of between those differentially expressed and those which aren't." width="700"><p class="caption">
Distribution of scores between differentially expressed genes and those
who aren’t. The line indicates the mean score of the similarity between
differentially expressed genes and those which aren’t differentially
expressed.
</p>
</div>
<p>Comparing the genes differentially expressed and those who aren’t
doesn’t show that they are non-randomly selected (p-value 0.7102897).
The genes with a p-value below the threshold are not more closely
functionally related than all the other genes<a class="footnote-ref" tabindex="0" data-bs-toggle="popover" data-bs-content="&lt;p&gt;From 400 genes there are 123 with pathway information
and only 24 where significantly differentially expressed in this
subset.&lt;/p&gt;"><sup>1</sup></a>.</p>
</div>
<div class="section level3">
<h3 id="are-functionally-related-the-selected-differentially-expressed-genes">Are functionally related the selected differentially expressed
genes?<a class="anchor" aria-label="anchor" href="#are-functionally-related-the-selected-differentially-expressed-genes"></a>
</h3>
<p>We have seen that the genes differentially expressed aren’t selected
by their functionality. However they could be more functionally related
than the other genes. Are the differentially expressed genes more
functionally similar than it would be expected ?</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="op">(</span><span class="va">scoreW</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/combineScores.html">combineScores</a></span><span class="op">(</span><span class="va">gS</span><span class="op">[</span><span class="va">keep</span>, <span class="va">keep</span><span class="op">]</span>, <span class="st">"avg"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">## [1] 0.1586497</span></span>
<span><span class="va">b</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/boot/man/boot.html" class="external-link">boot</a></span><span class="op">(</span>data <span class="op">=</span> <span class="va">gS</span>, R <span class="op">=</span> <span class="fl">1000</span>, statistic <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span>, <span class="va">i</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/mean.html" class="external-link">mean</a></span><span class="op">(</span><span class="va">x</span><span class="op">[</span><span class="va">i</span>, <span class="va">i</span><span class="op">]</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span></span>
<span><span class="op">(</span><span class="va">p.val</span> <span class="op">&lt;-</span> <span class="op">(</span><span class="fl">1</span> <span class="op">+</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="va">b</span><span class="op">$</span><span class="va">t</span> <span class="op">&gt;</span> <span class="va">scoreW</span><span class="op">)</span><span class="op">)</span> <span class="op">/</span> <span class="fl">1001</span><span class="op">)</span> <span class="co"># P-value</span></span>
<span><span class="co">## [1] 0.00999001</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/hist.html" class="external-link">hist</a></span><span class="op">(</span><span class="va">b</span><span class="op">$</span><span class="va">t</span>, main <span class="op">=</span> <span class="st">"Distribution of scores"</span>, xlab <span class="op">=</span> <span class="st">"Similarity score"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/abline.html" class="external-link">abline</a></span><span class="op">(</span>v <span class="op">=</span> <span class="va">scoreW</span>, col <span class="op">=</span> <span class="st">"red"</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="BioCor_2_advanced_files/figure-html/pval2-1.png" alt="Distribution of similarity scores between expressed genes. A vertical red line indicates the real similarity between those differentially expressed genes." width="700"><p class="caption">
Distribution of the similarity within differentially expressed genes.
The line indicates the mean funtional similarity whitin them.
</p>
</div>
<p>If we selected randomly the genes from our pool we would expect a
score around 0.1586497 with a probability of 0.00999. That means that
the differentially expressed genes is highly different compared to the
other genes if we use a significance threshold of 0.05 <a class="footnote-ref" tabindex="0" data-bs-toggle="popover" data-bs-content="&lt;p&gt;Remember that this is a small subset.&lt;/p&gt;"><sup>2</sup></a>.</p>
</div>
<div class="section level3">
<h3 id="influence-of-the-fold-change-in-the-functionally-similarity-of-the-genes">Influence of the fold change in the functionally similarity of the
genes<a class="anchor" aria-label="anchor" href="#influence-of-the-fold-change-in-the-functionally-similarity-of-the-genes"></a>
</h3>
<p>We have seen that the genes differentially expressed are not selected
by functional similarity but they are functionally related. Now we would
like to know if selecting a fold change threshold affects the functional
similarity between them.</p>
<p>To know the relationship between the fold change and the similarity
between genes we have several methods:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">s</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">max</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/MathFun.html" class="external-link">abs</a></span><span class="op">(</span><span class="va">subRes</span><span class="op">$</span><span class="va">log2FoldChange</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fl">0.05</span>, by <span class="op">=</span> <span class="fl">0.05</span><span class="op">)</span></span>
<span><span class="va">l</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">s</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="va">deg</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">subRes</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/MathFun.html" class="external-link">abs</a></span><span class="op">(</span><span class="va">subRes</span><span class="op">$</span><span class="va">log2FoldChange</span><span class="op">)</span> <span class="op">&gt;=</span> <span class="va">x</span>, <span class="op">]</span><span class="op">)</span></span>
<span>    <span class="va">keep</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">gS</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="va">genes</span><span class="op">[</span><span class="va">genes</span><span class="op">$</span><span class="va">ENSEMBL</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="va">deg</span>, <span class="st">"ENTREZID"</span><span class="op">]</span></span>
<span>    <span class="va">BetweenAbove</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/mean.html" class="external-link">mean</a></span><span class="op">(</span><span class="va">gS</span><span class="op">[</span><span class="va">keep</span>, <span class="va">keep</span><span class="op">]</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span>    <span class="va">AboveBelow</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/mean.html" class="external-link">mean</a></span><span class="op">(</span><span class="va">gS</span><span class="op">[</span><span class="va">keep</span>, <span class="op">!</span><span class="va">keep</span><span class="op">]</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span>    <span class="va">BetweenBelow</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/mean.html" class="external-link">mean</a></span><span class="op">(</span><span class="va">gS</span><span class="op">[</span><span class="op">!</span><span class="va">keep</span>, <span class="op">!</span><span class="va">keep</span><span class="op">]</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span>
<span>        <span class="st">"BetweenAbove"</span> <span class="op">=</span> <span class="va">BetweenAbove</span>, <span class="st">"AboveBelow"</span> <span class="op">=</span> <span class="va">AboveBelow</span>,</span>
<span>        <span class="st">"BetweenBelow"</span> <span class="op">=</span> <span class="va">BetweenBelow</span></span>
<span>    <span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span></span>
<span><span class="va">L</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/cbind.html" class="external-link">cbind</a></span><span class="op">(</span>logfc <span class="op">=</span> <span class="va">s</span>, <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/t.html" class="external-link">t</a></span><span class="op">(</span><span class="va">l</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">L</span><span class="op">$</span><span class="va">logfc</span>, <span class="va">L</span><span class="op">$</span><span class="va">BetweenAbove</span>,</span>
<span>    type <span class="op">=</span> <span class="st">"l"</span>, xlab <span class="op">=</span> <span class="st">"abs(log2) fold change"</span>,</span>
<span>    ylab <span class="op">=</span> <span class="st">"Similarity score"</span>,</span>
<span>    main <span class="op">=</span> <span class="st">"Similarity scores along logFC"</span>, col <span class="op">=</span> <span class="st">"darkred"</span></span>
<span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/lines.html" class="external-link">lines</a></span><span class="op">(</span><span class="va">L</span><span class="op">$</span><span class="va">logfc</span>, <span class="va">L</span><span class="op">$</span><span class="va">AboveBelow</span>, col <span class="op">=</span> <span class="st">"darkgreen"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/lines.html" class="external-link">lines</a></span><span class="op">(</span><span class="va">L</span><span class="op">$</span><span class="va">logfc</span>, <span class="va">L</span><span class="op">$</span><span class="va">BetweenBelow</span>, col <span class="op">=</span> <span class="st">"black"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/legend.html" class="external-link">legend</a></span><span class="op">(</span><span class="st">"topleft"</span>,</span>
<span>    legend <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span>
<span>        <span class="st">"Between genes above and below threshold"</span>,</span>
<span>        <span class="st">"Whitin genes above threshold"</span>,</span>
<span>        <span class="st">"Whitin genes below threshold"</span></span>
<span>    <span class="op">)</span>,</span>
<span>    fill <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"darkgreen"</span>, <span class="st">"darkred"</span>, <span class="st">"black"</span><span class="op">)</span></span>
<span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="BioCor_2_advanced_files/figure-html/logfc1-1.png" alt="A line plot on the X axis the absolute log fold change of the threshold used,  on the vertical axis the similarity score. Three lines, in red the similarity within  genes above the threshold, in black those below the threshold, in green between above  the threshold and below the threshold." width="700"><p class="caption">
Similarity of genes along abs(logFC). Assessing the similarity of genes
according to their absolute log2 fold change.
</p>
</div>
<p>The functional similarity of the genes above the threshold increases
with a more restrictive threshold, indicating that a logFC threshold
selects genes by their functionality, or in other words that genes
differentially expressed tend to be of related pathways. The similarity
between those genes above the threshold and below remains constant as
well as within genes below the threshold.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">l</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">s</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="co"># Names of genes up and down regulated</span></span>
<span>    <span class="va">degUp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">subRes</span><span class="op">[</span><span class="va">subRes</span><span class="op">$</span><span class="va">log2FoldChange</span> <span class="op">&gt;=</span> <span class="va">x</span>, <span class="op">]</span><span class="op">)</span></span>
<span>    <span class="va">degDown</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">subRes</span><span class="op">[</span><span class="va">subRes</span><span class="op">$</span><span class="va">log2FoldChange</span> <span class="op">&lt;=</span> <span class="op">-</span><span class="va">x</span>, <span class="op">]</span><span class="op">)</span></span>
<span></span>
<span>    <span class="co"># Translate to ids in gS</span></span>
<span>    <span class="va">keepUp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">gS</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="va">genes</span><span class="op">[</span><span class="va">genes</span><span class="op">$</span><span class="va">ENSEMBL</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="va">degUp</span>, <span class="st">"ENTREZID"</span><span class="op">]</span></span>
<span>    <span class="va">keepDown</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">gS</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="va">genes</span><span class="op">[</span><span class="va">genes</span><span class="op">$</span><span class="va">ENSEMBL</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="va">degDown</span>, <span class="st">"ENTREZID"</span><span class="op">]</span></span>
<span></span>
<span>    <span class="co"># Calculate the mean similarity between each subgrup</span></span>
<span>    <span class="va">between</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/mean.html" class="external-link">mean</a></span><span class="op">(</span><span class="va">gS</span><span class="op">[</span><span class="va">keepUp</span>, <span class="va">keepDown</span><span class="op">]</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"UpVsDown"</span> <span class="op">=</span> <span class="va">between</span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span></span>
<span><span class="va">L</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/cbind.html" class="external-link">cbind</a></span><span class="op">(</span><span class="st">"logfc"</span> <span class="op">=</span> <span class="va">s</span>, <span class="st">"UpVsDown"</span> <span class="op">=</span> <span class="va">l</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="va">L</span><span class="op">$</span><span class="va">logfc</span>, <span class="va">L</span><span class="op">$</span><span class="va">UpVsDown</span>,</span>
<span>    type <span class="op">=</span> <span class="st">"l"</span>,</span>
<span>    xlab <span class="op">=</span> <span class="st">"abs(log2) fold change threshold"</span>,</span>
<span>    ylab <span class="op">=</span> <span class="st">"Similarity score"</span>,</span>
<span>    main <span class="op">=</span> <span class="st">"Similarity scores along logFC"</span></span>
<span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/legend.html" class="external-link">legend</a></span><span class="op">(</span><span class="st">"topright"</span>,</span>
<span>    legend <span class="op">=</span> <span class="st">"Up vs down regulated genes"</span>,</span>
<span>    fill <span class="op">=</span> <span class="st">"black"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="BioCor_2_advanced_files/figure-html/logfc2-1.png" alt="Similarity score between genes up and down-regulated along the threshold of log fold change." width="700"><p class="caption">
Functional similarity between the up-regulated and down-regulated genes.
</p>
</div>
<p>The maximal functional similarity between genes up-regulated and
down-regulated are at 1.15 log2 fold change.</p>
</div>
</div>
<div class="section level2">
<h2 id="assessing-a-new-pathway">Assessing a new pathway<a class="anchor" aria-label="anchor" href="#assessing-a-new-pathway"></a>
</h2>
<p>Sometimes the top differentially expressed genes or some other key
genes are selected as a signature or a potential new group of related
genes. In those cases we can test how does the network of genes change
if we add them. Here we create a new pathway named <code>deg</code>, and
we see the effect on the functional similarity score for all the
genes:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Adding a new pathway "deg" to those genes</span></span>
<span><span class="va">genesReact2</span> <span class="op">&lt;-</span> <span class="va">genesReact</span></span>
<span><span class="va">diffGenes</span> <span class="op">&lt;-</span> <span class="va">genes</span><span class="op">[</span><span class="va">genes</span><span class="op">$</span><span class="va">ENSEMBL</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="va">deg</span>, <span class="st">"ENTREZID"</span><span class="op">]</span></span>
<span><span class="va">genesReact2</span><span class="op">[</span><span class="va">diffGenes</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">genesReact</span><span class="op">[</span><span class="va">diffGenes</span><span class="op">]</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">x</span>, <span class="st">"deg"</span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/plot.default.html" class="external-link">plot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/ecdf.html" class="external-link">ecdf</a></span><span class="op">(</span><span class="fu"><a href="../reference/mgeneSim.html">mgeneSim</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">genesReact</span><span class="op">)</span>, <span class="va">genesReact</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/curve.html" class="external-link">curve</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/ecdf.html" class="external-link">ecdf</a></span><span class="op">(</span><span class="fu"><a href="../reference/mgeneSim.html">mgeneSim</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">genesReact2</span><span class="op">)</span>, <span class="va">genesReact2</span><span class="op">)</span><span class="op">)</span>, color <span class="op">=</span> <span class="st">"red"</span><span class="op">)</span></span></code></pre></div>
<p>This would take lot of time, for a illustration purpose we reduce to
some genes:</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://hbiostat.org/R/Hmisc/" class="external-link">"Hmisc"</a></span><span class="op">)</span></span>
<span><span class="va">genesReact2</span> <span class="op">&lt;-</span> <span class="va">genesReact</span></span>
<span><span class="va">diffGenes</span> <span class="op">&lt;-</span> <span class="va">genes</span><span class="op">[</span><span class="va">genes</span><span class="op">$</span><span class="va">ENSEMBL</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="va">deg</span>, <span class="st">"ENTREZID"</span><span class="op">]</span></span>
<span><span class="co"># Create the new pathway called deg</span></span>
<span><span class="va">genesReact2</span><span class="op">[</span><span class="va">diffGenes</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">genesReact</span><span class="op">[</span><span class="va">diffGenes</span><span class="op">]</span>, <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">x</span>, <span class="st">"deg"</span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span></span>
<span><span class="va">ids</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">genes</span><span class="op">[</span><span class="va">genes</span><span class="op">$</span><span class="va">ENSEMBL</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">subRes</span><span class="op">)</span>, <span class="st">"ENTREZID"</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/Hmisc/man/Ecdf.html" class="external-link">Ecdf</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="../reference/mgeneSim.html">mgeneSim</a></span><span class="op">(</span><span class="va">ids</span>, <span class="va">genesReact</span>, method <span class="op">=</span> <span class="st">"BMA"</span><span class="op">)</span>,</span>
<span>       <span class="fu"><a href="../reference/mgeneSim.html">mgeneSim</a></span><span class="op">(</span><span class="va">ids</span>, <span class="va">genesReact2</span>, method <span class="op">=</span> <span class="st">"BMA"</span><span class="op">)</span></span>
<span><span class="op">)</span>,</span>
<span>group <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"Reactome"</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">ids</span><span class="op">)</span><span class="op">^</span><span class="fl">2</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"Modified"</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">ids</span><span class="op">)</span><span class="op">^</span><span class="fl">2</span><span class="op">)</span><span class="op">)</span>,</span>
<span>col <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"black"</span>, <span class="st">"red"</span><span class="op">)</span>, xlab <span class="op">=</span> <span class="st">"Functional similarities"</span>, </span>
<span>main <span class="op">=</span> <span class="st">"Empirical cumulative distribution"</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="BioCor_2_advanced_files/figure-html/newPathway2-1.png" alt="Empirical cumulative distribution of the functional similarity with the original data  (colored in red) and with an added pathway (in black)." width="700"><p class="caption">
The effect of adding a new pathway to a functional similarity. In red
the same network as in black but with the added pathway.
</p>
</div>
</div>
<div class="section level2">
<h2 id="merging-sources-of-information">Merging sources of information<a class="anchor" aria-label="anchor" href="#merging-sources-of-information"></a>
</h2>
<p>Sometimes we have several origin of information, either several
databases, or information from other programs… We can merge this in the
single object required by the function in <code>BioCor</code> using the
function <code>combineSources</code><a class="footnote-ref" tabindex="0" data-bs-toggle="popover" data-bs-content="&lt;p&gt;See the help page of &lt;code&gt;combineSources&lt;/code&gt;&lt;/p&gt;"><sup>3</sup></a>.</p>
<p>This functions helps to evaluate what happens when we add more
pathway information. For instance here we add the information in Kegg
and the information in Reactome and we visualize it using the same
procedure as previously:</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">genesKegg</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/as.list.html" class="external-link">as.list</a></span><span class="op">(</span><span class="va">org.Hs.egPATH</span><span class="op">)</span></span>
<span><span class="va">gSK</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/mgeneSim.html">mgeneSim</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">gS</span><span class="op">)</span>, <span class="va">genesKegg</span><span class="op">)</span></span>
<span><span class="va">mix</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/combineSources.html">combineSources</a></span><span class="op">(</span><span class="va">genesKegg</span>, <span class="va">genesReact</span><span class="op">)</span></span>
<span><span class="co">## Warning in combineSources(genesKegg, genesReact): More than 50% of genes identifiers of a source are unique</span></span>
<span><span class="co">## Check the identifiers of the genes</span></span>
<span><span class="va">gSMix</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/mgeneSim.html">mgeneSim</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">gS</span><span class="op">)</span>, <span class="va">mix</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/Hmisc/man/Ecdf.html" class="external-link">Ecdf</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">gS</span>, <span class="va">gSK</span>, <span class="va">gSMix</span><span class="op">)</span>,</span>
<span>    group <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span>
<span>        <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"Reactome"</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">gS</span><span class="op">)</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"Kegg"</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">gSK</span><span class="op">)</span><span class="op">)</span>,</span>
<span>        <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"Mix"</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">gSMix</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="op">)</span>,</span>
<span>    col <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"black"</span>, <span class="st">"red"</span>, <span class="st">"blue"</span><span class="op">)</span>, xlab <span class="op">=</span> <span class="st">"Functional similarities"</span>, </span>
<span>    main <span class="op">=</span> <span class="st">"Empirical cumulative distribution."</span></span>
<span><span class="op">)</span></span>
<span><span class="co">## Warning in regularize.values(x, y, ties, missing(ties), na.rm = na.rm):</span></span>
<span><span class="co">## collapsing to unique 'x' values</span></span>
<span><span class="co">## Warning in regularize.values(x, y, ties, missing(ties), na.rm = na.rm):</span></span>
<span><span class="co">## collapsing to unique 'x' values</span></span>
<span></span>
<span><span class="co">## Warning in regularize.values(x, y, ties, missing(ties), na.rm = na.rm):</span></span>
<span><span class="co">## collapsing to unique 'x' values</span></span></code></pre></div>
<div class="figure">
<img src="BioCor_2_advanced_files/figure-html/combineSource-1.png" alt="Comparing the functional similarity by looking at the empirical cumulative distribution. Kegg in black, Reactome in blue, and both mixed in red." width="700"><p class="caption">
Comparison of functional similarity in different databases.
</p>
</div>
<p>When mixed, there is a huge increase in the genes that share a
pathway using the <code>max</code> method. Observe in next figure
(@ref(fig:combineSource2)) how does the method affects to the
results:</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gSK2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/mgeneSim.html">mgeneSim</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">gS</span><span class="op">)</span>, <span class="va">genesKegg</span>, method <span class="op">=</span> <span class="st">"BMA"</span><span class="op">)</span></span>
<span><span class="va">gS2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/mgeneSim.html">mgeneSim</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">gS</span><span class="op">)</span>, <span class="va">genesReact</span>, method <span class="op">=</span> <span class="st">"BMA"</span><span class="op">)</span></span>
<span><span class="va">gSMix2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/mgeneSim.html">mgeneSim</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">gS</span><span class="op">)</span>, <span class="va">mix</span>, method <span class="op">=</span> <span class="st">"BMA"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/Hmisc/man/Ecdf.html" class="external-link">Ecdf</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">gS2</span>, <span class="va">gSK2</span>, <span class="va">gSMix2</span><span class="op">)</span>,</span>
<span>    group <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span>
<span>        <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"Reactome"</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">gS</span><span class="op">)</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"Kegg"</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">gSK</span><span class="op">)</span><span class="op">)</span>,</span>
<span>        <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"Mix"</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">gSMix</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="op">)</span>,</span>
<span>    col <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"black"</span>, <span class="st">"red"</span>, <span class="st">"blue"</span><span class="op">)</span>, xlab <span class="op">=</span> <span class="st">"Functional similarities (BMA)"</span>, main <span class="op">=</span> <span class="st">"Empirical cumulative distribution."</span></span>
<span><span class="op">)</span></span>
<span><span class="co">## Warning in regularize.values(x, y, ties, missing(ties), na.rm = na.rm):</span></span>
<span><span class="co">## collapsing to unique 'x' values</span></span>
<span></span>
<span><span class="co">## Warning in regularize.values(x, y, ties, missing(ties), na.rm = na.rm):</span></span>
<span><span class="co">## collapsing to unique 'x' values</span></span>
<span></span>
<span><span class="co">## Warning in regularize.values(x, y, ties, missing(ties), na.rm = na.rm):</span></span>
<span><span class="co">## collapsing to unique 'x' values</span></span></code></pre></div>
<div class="figure">
<img src="BioCor_2_advanced_files/figure-html/combineSource2-1.png" alt="Empirical cumulative distribution of pathways according to  Kegg (black), Reactome (blue) and a mix of both (red)." width="700"><p class="caption">
Comparison of functional similarity in different gene sets.
</p>
</div>
<p>Now we can appreciate that most of the functional similarity is
brought by Reactome database</p>
</div>
<div class="section level2">
<h2 id="mirna-analysis">miRNA analysis<a class="anchor" aria-label="anchor" href="#mirna-analysis"></a>
</h2>
<p>miRNAs are RNAs that interact with many genes and transcripts and is
subject to change with location and time, thus defining the effect of an
miRNA is difficult. In this section we try to answer how functionally
similar are miRNA between them to provide a help for potentially closely
related miRNA.</p>
<p>First we look for human miRNAs and prepare them for using them as the
input for the cluster functions (Restricting to 10 miRNA with less than
150 genes).<a class="footnote-ref" tabindex="0" data-bs-toggle="popover" data-bs-content='&lt;p&gt;This preparation has been adapted from a &lt;a href="https://support.bioconductor.org/p/48138/#48236" class="external-link"&gt;previous
discussion&lt;/a&gt;.&lt;/p&gt;'><sup>4</sup></a></p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st">"targetscan.Hs.eg.db"</span><span class="op">)</span></span>
<span><span class="co">## select human mirna</span></span>
<span><span class="va">humanMirnaIdx</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/grep.html" class="external-link">grep</a></span><span class="op">(</span><span class="st">"hsa"</span>, <span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/Bimap-keys.html" class="external-link">mappedkeys</a></span><span class="op">(</span><span class="va">targetscan.Hs.egMIRNA</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">## select seed-based families for human mirna</span></span>
<span><span class="va">humanMirna</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/Bimap-keys.html" class="external-link">mappedkeys</a></span><span class="op">(</span><span class="va">targetscan.Hs.egMIRNA</span><span class="op">)</span><span class="op">[</span><span class="va">humanMirnaIdx</span><span class="op">]</span></span>
<span><span class="co">## select targets of families</span></span>
<span><span class="va">humanMirnaFamilies</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/unlist.html" class="external-link">unlist</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/get.html" class="external-link">mget</a></span><span class="op">(</span><span class="va">humanMirna</span>, <span class="va">targetscan.Hs.egMIRBASE2FAMILY</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">humanMirnaTargets</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/get.html" class="external-link">mget</a></span><span class="op">(</span><span class="va">humanMirnaFamilies</span>, <span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/Bimap-direction.html" class="external-link">revmap</a></span><span class="op">(</span><span class="va">targetscan.Hs.egTARGETS</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">humanMirnaTargets</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="va">humanMirna</span></span>
<span><span class="co"># Restrict to miRNA with more than one target and less than 150</span></span>
<span><span class="va">miRNAs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/AnnotationDbi/man/Bimap-envirAPI.html" class="external-link">sample</a></span><span class="op">(</span><span class="va">humanMirnaTargets</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/lengths.html" class="external-link">lengths</a></span><span class="op">(</span><span class="va">humanMirnaTargets</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">1</span> <span class="op">&amp;</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/lengths.html" class="external-link">lengths</a></span><span class="op">(</span><span class="va">humanMirnaTargets</span><span class="op">)</span> <span class="op">&lt;</span> <span class="fl">150</span><span class="op">]</span>, <span class="fl">10</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/lengths.html" class="external-link">lengths</a></span><span class="op">(</span><span class="va">miRNAs</span><span class="op">)</span></span>
<span><span class="co">##    hsa-miR-187 hsa-miR-127-3p   hsa-miR-551a    hsa-miR-191 hsa-miR-615-3p </span></span>
<span><span class="co">##             13             17             17            118             28 </span></span>
<span><span class="co">##    hsa-miR-99b   hsa-miR-551b   hsa-miR-487b    hsa-miR-184    hsa-miR-451 </span></span>
<span><span class="co">##             92             17             23             54             36</span></span></code></pre></div>
<p>Now we calculate the functional similarity of those miRNAs using a
cluster approach.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">cluster1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/mclusterSim.html">mclusterSim</a></span><span class="op">(</span><span class="va">miRNAs</span>, <span class="va">genesReact</span>, method <span class="op">=</span> <span class="st">"BMA"</span><span class="op">)</span></span>
<span><span class="co">## Warning in mclusterSim(miRNAs, genesReact, method = "BMA"): Some genes are not</span></span>
<span><span class="co">## in the list provided.</span></span>
<span><span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/Round.html" class="external-link">round</a></span><span class="op">(</span><span class="va">cluster1</span>, <span class="fl">2</span><span class="op">)</span>, caption <span class="op">=</span> <span class="st">"The similarity between miRNA"</span>, format <span class="op">=</span> <span class="st">"html"</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<caption>
The similarity between miRNA
</caption>
<thead><tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
hsa-miR-187
</th>
<th style="text-align:right;">
hsa-miR-127-3p
</th>
<th style="text-align:right;">
hsa-miR-551a
</th>
<th style="text-align:right;">
hsa-miR-191
</th>
<th style="text-align:right;">
hsa-miR-615-3p
</th>
<th style="text-align:right;">
hsa-miR-99b
</th>
<th style="text-align:right;">
hsa-miR-551b
</th>
<th style="text-align:right;">
hsa-miR-487b
</th>
<th style="text-align:right;">
hsa-miR-184
</th>
<th style="text-align:right;">
hsa-miR-451
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
hsa-miR-187
</td>
<td style="text-align:right;">
1.00
</td>
<td style="text-align:right;">
0.52
</td>
<td style="text-align:right;">
0.63
</td>
<td style="text-align:right;">
0.63
</td>
<td style="text-align:right;">
0.57
</td>
<td style="text-align:right;">
0.67
</td>
<td style="text-align:right;">
0.63
</td>
<td style="text-align:right;">
0.62
</td>
<td style="text-align:right;">
0.68
</td>
<td style="text-align:right;">
0.49
</td>
</tr>
<tr>
<td style="text-align:left;">
hsa-miR-127-3p
</td>
<td style="text-align:right;">
0.52
</td>
<td style="text-align:right;">
1.00
</td>
<td style="text-align:right;">
0.41
</td>
<td style="text-align:right;">
0.48
</td>
<td style="text-align:right;">
0.42
</td>
<td style="text-align:right;">
0.45
</td>
<td style="text-align:right;">
0.41
</td>
<td style="text-align:right;">
0.44
</td>
<td style="text-align:right;">
0.42
</td>
<td style="text-align:right;">
0.41
</td>
</tr>
<tr>
<td style="text-align:left;">
hsa-miR-551a
</td>
<td style="text-align:right;">
0.63
</td>
<td style="text-align:right;">
0.41
</td>
<td style="text-align:right;">
1.00
</td>
<td style="text-align:right;">
0.67
</td>
<td style="text-align:right;">
0.82
</td>
<td style="text-align:right;">
0.62
</td>
<td style="text-align:right;">
1.00
</td>
<td style="text-align:right;">
0.78
</td>
<td style="text-align:right;">
0.62
</td>
<td style="text-align:right;">
0.55
</td>
</tr>
<tr>
<td style="text-align:left;">
hsa-miR-191
</td>
<td style="text-align:right;">
0.63
</td>
<td style="text-align:right;">
0.48
</td>
<td style="text-align:right;">
0.67
</td>
<td style="text-align:right;">
1.00
</td>
<td style="text-align:right;">
0.71
</td>
<td style="text-align:right;">
0.78
</td>
<td style="text-align:right;">
0.67
</td>
<td style="text-align:right;">
0.78
</td>
<td style="text-align:right;">
0.72
</td>
<td style="text-align:right;">
0.73
</td>
</tr>
<tr>
<td style="text-align:left;">
hsa-miR-615-3p
</td>
<td style="text-align:right;">
0.57
</td>
<td style="text-align:right;">
0.42
</td>
<td style="text-align:right;">
0.82
</td>
<td style="text-align:right;">
0.71
</td>
<td style="text-align:right;">
1.00
</td>
<td style="text-align:right;">
0.65
</td>
<td style="text-align:right;">
0.82
</td>
<td style="text-align:right;">
0.78
</td>
<td style="text-align:right;">
0.61
</td>
<td style="text-align:right;">
0.61
</td>
</tr>
<tr>
<td style="text-align:left;">
hsa-miR-99b
</td>
<td style="text-align:right;">
0.67
</td>
<td style="text-align:right;">
0.45
</td>
<td style="text-align:right;">
0.62
</td>
<td style="text-align:right;">
0.78
</td>
<td style="text-align:right;">
0.65
</td>
<td style="text-align:right;">
1.00
</td>
<td style="text-align:right;">
0.62
</td>
<td style="text-align:right;">
0.75
</td>
<td style="text-align:right;">
0.76
</td>
<td style="text-align:right;">
0.73
</td>
</tr>
<tr>
<td style="text-align:left;">
hsa-miR-551b
</td>
<td style="text-align:right;">
0.63
</td>
<td style="text-align:right;">
0.41
</td>
<td style="text-align:right;">
1.00
</td>
<td style="text-align:right;">
0.67
</td>
<td style="text-align:right;">
0.82
</td>
<td style="text-align:right;">
0.62
</td>
<td style="text-align:right;">
1.00
</td>
<td style="text-align:right;">
0.78
</td>
<td style="text-align:right;">
0.62
</td>
<td style="text-align:right;">
0.55
</td>
</tr>
<tr>
<td style="text-align:left;">
hsa-miR-487b
</td>
<td style="text-align:right;">
0.62
</td>
<td style="text-align:right;">
0.44
</td>
<td style="text-align:right;">
0.78
</td>
<td style="text-align:right;">
0.78
</td>
<td style="text-align:right;">
0.78
</td>
<td style="text-align:right;">
0.75
</td>
<td style="text-align:right;">
0.78
</td>
<td style="text-align:right;">
1.00
</td>
<td style="text-align:right;">
0.67
</td>
<td style="text-align:right;">
0.66
</td>
</tr>
<tr>
<td style="text-align:left;">
hsa-miR-184
</td>
<td style="text-align:right;">
0.68
</td>
<td style="text-align:right;">
0.42
</td>
<td style="text-align:right;">
0.62
</td>
<td style="text-align:right;">
0.72
</td>
<td style="text-align:right;">
0.61
</td>
<td style="text-align:right;">
0.76
</td>
<td style="text-align:right;">
0.62
</td>
<td style="text-align:right;">
0.67
</td>
<td style="text-align:right;">
1.00
</td>
<td style="text-align:right;">
0.67
</td>
</tr>
<tr>
<td style="text-align:left;">
hsa-miR-451
</td>
<td style="text-align:right;">
0.49
</td>
<td style="text-align:right;">
0.41
</td>
<td style="text-align:right;">
0.55
</td>
<td style="text-align:right;">
0.73
</td>
<td style="text-align:right;">
0.61
</td>
<td style="text-align:right;">
0.73
</td>
<td style="text-align:right;">
0.55
</td>
<td style="text-align:right;">
0.66
</td>
<td style="text-align:right;">
0.67
</td>
<td style="text-align:right;">
1.00
</td>
</tr>
</tbody>
</table>
<p>So for instance hsa-miR-551b, hsa-miR-551a are functionally related
despite being from different families.</p>
</div>
<div class="section level2">
<h2 id="comparing-with-go-similarities">Comparing with GO similarities<a class="anchor" aria-label="anchor" href="#comparing-with-go-similarities"></a>
</h2>
<p>As suggested in the main vignette functional similarities can be
compared to semantic similarities such as those based on GO. Here a
comparison using the biological process from the gene ontologies is
done:</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://yulab-smu.top/biomedical-knowledge-mining-book/" class="external-link">"GOSemSim"</a></span><span class="op">)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## GOSemSim v2.26.0  For help: https://yulab-smu.top/biomedical-knowledge-mining-book/</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## If you use GOSemSim in published research, please cite:</span></span>
<span><span class="co">## <span style="color: #00BBBB;">-</span> Guangchuang Yu. Gene Ontology Semantic Similarity Analysis Using GOSemSim. In: Kidder B. (eds) Stem Cell Transcriptional Networks. Methods in Molecular Biology, 2020, 2117:207-215. Humana, New York, NY. doi:10.1007/978-1-0716-0301-7_11</span></span>
<span><span class="co">## <span style="color: #00BBBB;">-</span> Guangchuang Yu, Fei Li, Yide Qin, Xiaochen Bo, Yibo Wu, Shengqi Wang. GOSemSim: an R package for measuring semantic similarity among GO terms and gene products Bioinformatics 2010, 26(7):976-978. doi:10.1093/bioinformatics/btq064</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Attaching package: 'GOSemSim'</span></span>
<span><span class="co">## The following objects are masked from 'package:BioCor':</span></span>
<span><span class="co">## </span></span>
<span><span class="co">##     clusterSim, combineScores, geneSim, mclusterSim, mgeneSim</span></span>
<span><span class="va">BP</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GOSemSim/man/godata.html" class="external-link">godata</a></span><span class="op">(</span><span class="st">"org.Hs.eg.db"</span>, ont <span class="op">=</span> <span class="st">"BP"</span>, computeIC <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">## preparing gene to GO mapping data...</span></span>
<span><span class="co">## preparing IC data...</span></span>
<span><span class="va">gsGO</span> <span class="op">&lt;-</span> <span class="fu">GOSemSim</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/GOSemSim/man/mgeneSim.html" class="external-link">mgeneSim</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">gS</span><span class="op">)</span>, semData <span class="op">=</span> <span class="va">BP</span>, measure <span class="op">=</span> <span class="st">"Resnik"</span>, verbose <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="va">keep</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">gS</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/BiocGenerics/man/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">gsGO</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/graphics/hist.html" class="external-link">hist</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/dist.html" class="external-link">as.dist</a></span><span class="op">(</span><span class="va">gS</span><span class="op">[</span><span class="va">keep</span>, <span class="va">keep</span><span class="op">]</span> <span class="op">-</span> <span class="va">gsGO</span><span class="op">)</span>,</span>
<span>    main <span class="op">=</span> <span class="st">"Difference between functional similarity and biological process"</span>,</span>
<span>    xlab <span class="op">=</span> <span class="st">"Functional similarity - biological process similarity"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="BioCor_2_advanced_files/figure-html/GOSemSim-1.png" alt="Functional similarities for the same genes compared between GO and Reactome annotation." width="700"><p class="caption">
Comparison of similarities. Functional similarities compared to
biological process semantic similarity.
</p>
</div>
<p>On this graphic we can observe that some genes have a large
functional similarity and few biological similarity. They are present
together in several pathways while they share few biological process,
indicating that they might be key elements of the pathways they are in.
On the other hand some other pairs of genes show higher biological
process similarity than functional similarity, indicating specialization
or compartmentalization of said genes.</p>
</div>
<div class="section level2">
<h2 class="unnumbered" id="session-info">Session Info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">## R version 4.3.0 (2023-04-21)</span></span>
<span><span class="co">## Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span><span class="co">## Running under: Ubuntu 22.04.2 LTS</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 </span></span>
<span><span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">##  [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8       </span></span>
<span><span class="co">##  [4] LC_COLLATE=C.UTF-8     LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8   </span></span>
<span><span class="co">##  [7] LC_PAPER=C.UTF-8       LC_NAME=C              LC_ADDRESS=C          </span></span>
<span><span class="co">## [10] LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## time zone: UTC</span></span>
<span><span class="co">## tzcode source: system (glibc)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">## [8] base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">##  [1] GOSemSim_2.26.0             targetscan.Hs.eg.db_0.6.1  </span></span>
<span><span class="co">##  [3] Hmisc_5.0-1                 boot_1.3-28.1              </span></span>
<span><span class="co">##  [5] DESeq2_1.40.0               airway_1.20.0              </span></span>
<span><span class="co">##  [7] SummarizedExperiment_1.30.0 GenomicRanges_1.52.0       </span></span>
<span><span class="co">##  [9] GenomeInfoDb_1.36.0         MatrixGenerics_1.12.0      </span></span>
<span><span class="co">## [11] matrixStats_0.63.0          BioCor_1.25.1              </span></span>
<span><span class="co">## [13] org.Hs.eg.db_3.17.0         AnnotationDbi_1.62.0       </span></span>
<span><span class="co">## [15] IRanges_2.34.0              S4Vectors_0.38.0           </span></span>
<span><span class="co">## [17] Biobase_2.60.0              BiocGenerics_0.46.0        </span></span>
<span><span class="co">## [19] BiocStyle_2.28.0           </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##  [1] DBI_1.1.3               bitops_1.0-7            gridExtra_2.3          </span></span>
<span><span class="co">##  [4] GSEABase_1.62.0         rlang_1.1.0             magrittr_2.0.3         </span></span>
<span><span class="co">##  [7] compiler_4.3.0          RSQLite_2.3.1           png_0.1-8              </span></span>
<span><span class="co">## [10] systemfonts_1.0.4       vctrs_0.6.2             reactome.db_1.84.0     </span></span>
<span><span class="co">## [13] stringr_1.5.0           pkgconfig_2.0.3         crayon_1.5.2           </span></span>
<span><span class="co">## [16] fastmap_1.1.1           backports_1.4.1         XVector_0.40.0         </span></span>
<span><span class="co">## [19] utf8_1.2.3              rmarkdown_2.21          graph_1.78.0           </span></span>
<span><span class="co">## [22] ragg_1.2.5              purrr_1.0.1             bit_4.0.5              </span></span>
<span><span class="co">## [25] xfun_0.39               zlibbioc_1.46.0         cachem_1.0.7           </span></span>
<span><span class="co">## [28] jsonlite_1.8.4          blob_1.2.4              highr_0.10             </span></span>
<span><span class="co">## [31] DelayedArray_0.25.0     BiocParallel_1.34.0     parallel_4.3.0         </span></span>
<span><span class="co">## [34] cluster_2.1.4           R6_2.5.1                bslib_0.4.2            </span></span>
<span><span class="co">## [37] stringi_1.7.12          rpart_4.1.19            jquerylib_0.1.4        </span></span>
<span><span class="co">## [40] Rcpp_1.0.10             bookdown_0.33           knitr_1.42             </span></span>
<span><span class="co">## [43] base64enc_0.1-3         Matrix_1.5-4            nnet_7.3-18            </span></span>
<span><span class="co">## [46] rstudioapi_0.14         yaml_2.3.7              codetools_0.2-19       </span></span>
<span><span class="co">## [49] lattice_0.21-8          tibble_3.2.1            KEGGREST_1.40.0        </span></span>
<span><span class="co">## [52] evaluate_0.20           foreign_0.8-84          desc_1.4.2             </span></span>
<span><span class="co">## [55] Biostrings_2.68.0       pillar_1.9.0            BiocManager_1.30.20    </span></span>
<span><span class="co">## [58] checkmate_2.1.0         rprojroot_2.0.3         RCurl_1.98-1.12        </span></span>
<span><span class="co">## [61] ggplot2_3.4.2           munsell_0.5.0           scales_1.2.1           </span></span>
<span><span class="co">## [64] xtable_1.8-4            glue_1.6.2              tools_4.3.0            </span></span>
<span><span class="co">## [67] data.table_1.14.8       annotate_1.78.0         locfit_1.5-9.7         </span></span>
<span><span class="co">## [70] fs_1.6.2                XML_3.99-0.14           grid_4.3.0             </span></span>
<span><span class="co">## [73] colorspace_2.1-0        GenomeInfoDbData_1.2.10 htmlTable_2.4.1        </span></span>
<span><span class="co">## [76] Formula_1.2-5           cli_3.6.1               textshaping_0.3.6      </span></span>
<span><span class="co">## [79] fansi_1.0.4             gtable_0.3.3            sass_0.4.5             </span></span>
<span><span class="co">## [82] digest_0.6.31           htmlwidgets_1.6.2       memoise_2.0.1          </span></span>
<span><span class="co">## [85] htmltools_0.5.5         pkgdown_2.0.7           lifecycle_1.0.3        </span></span>
<span><span class="co">## [88] httr_1.4.5              GO.db_3.17.0            bit64_4.0.5</span></span></code></pre></div>
</div>

  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Lluís Revilla Sancho.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
