<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Advanced usage of BioCor • BioCor</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js" integrity="sha256-FgpCb/KJQlLNfOu91ta32o/NMZxltwRo8QtmkMRdAu8=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha256-916EbMg70RQy9LHiGkXzG8hSg9EdNy97GazNG/aiY1w=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha256-U5ZEeKfGNOja007MMD3YBI0A3OSZOQbeG6z2f2Y0hu8=" crossorigin="anonymous"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/all.min.css" integrity="sha256-nAmazAk6vS34Xqo0BSrTb+abbtFlgsFK7NKSi6o7Y78=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.7.1/css/v4-shims.min.css" integrity="sha256-6qHlizsOWFskGlwVOKuns+D1nB6ssZrHQrNj1wGplHc=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.4/clipboard.min.js" integrity="sha256-FiZwavyI2V6+EXO1U+xzLG3IKldpiTFf3153ea9zikQ=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/headroom.min.js" integrity="sha256-DJFC1kqIhelURkuza0AvYal5RxMtpzLjFhsnVIeuk+U=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.9.4/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Advanced usage of BioCor">
<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">BioCor</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">1.9.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/BioCor_1_basics.html">About BioCor</a>
    </li>
    <li>
      <a href="../articles/BioCor_2_advanced.html">Advanced usage of BioCor</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/llrs/BioCor">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1>Advanced usage of BioCor</h1>
                        <h4 class="author">Lluís Revilla</h4>
            <address class="author_afil">
      August Pi i Sunyer Biomedical Research Institute (IDIBAPS); Liver Unit, Hospital Clinic<br><a class="author_email" href="mailto:#"></a><a href="mailto:lrevilla@clinic.cat">lrevilla@clinic.cat</a>
      </address>
                              <h4 class="author">Juan José Lozano</h4>
            <address class="author_afil">
      Centro de Investigación Biomédica en Red de Enfermedades Hepaticas y Digestivas (CIBEREHD); Barcelona, Spain<br><h4 class="author">Pau Sancho-Bru</h4>
            <address class="author_afil">
      August Pi i Sunyer Biomedical Research Institute (IDIBAPS); Liver Unit, Hospital Clinic<br><h4 class="date">4 October 2019</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/llrs/BioCor/blob/master/vignettes/BioCor_2_advanced.Rmd"><code>vignettes/BioCor_2_advanced.Rmd</code></a></small>
      <div class="hidden name"><code>BioCor_2_advanced.Rmd</code></div>

    </address>
</address>
</div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      Describes how to use the BioCor package to answer several biological questions and how to use functional similarities with other measures.
    </div>
    
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p>In this vignette we assume that the reader is already familiarized with the <a href="https://bioconductor.org/packages/3.10/BioCor/vignettes/BioCor_1_basics.html">introduction vignette</a>, but wants to know how can it help to answer other questions in other situations</p>
<p>We follow the same convention about names of the objects used <code>genesReact</code> and <code>genesKegg</code> are the list with information about the pathways the genes are involved in.</p>
</div>
<div id="merging" class="section level1">
<h1 class="hasAnchor">
<a href="#merging" class="anchor"></a>Merging similarities</h1>
<p>If one calculates similarities with KEGG data and Reactome or other input for the same genes or clusters BioCor provides a couple of functions to merge them.</p>
<p>We can set a weight to each similarity input with <code>weighted.sum</code>, multiply them also using a weight for each similarity (with <code>weighted.prod</code>), doing the mean or just adding them up. Similarities allow us to apply a function to combine the matrices of a list. Here we use some of the genes used in the first vignette:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" data-line-number="1">kegg &lt;-<span class="st"> </span><span class="kw"><a href="../reference/mgeneSim.html">mgeneSim</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"672"</span>, <span class="st">"675"</span>, <span class="st">"10"</span>), genesKegg)</a>
<a class="sourceLine" id="cb1-2" data-line-number="2">react &lt;-<span class="st"> </span><span class="kw"><a href="../reference/mgeneSim.html">mgeneSim</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"672"</span>, <span class="st">"675"</span>, <span class="st">"10"</span>), genesReact)</a>
<a class="sourceLine" id="cb1-3" data-line-number="3">## We can sum it adding a weight to each origin</a>
<a class="sourceLine" id="cb1-4" data-line-number="4"><span class="kw"><a href="../reference/weighted.html">weighted.sum</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(kegg[<span class="st">"672"</span>, <span class="st">"675"</span>], react[<span class="st">"672"</span>,<span class="st">"675"</span>]), <span class="dt">w =</span> <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">0.3</span>, <span class="fl">0.7</span>))</a>
<a class="sourceLine" id="cb1-5" data-line-number="5">## [1] 0.7247289</a>
<a class="sourceLine" id="cb1-6" data-line-number="6"></a>
<a class="sourceLine" id="cb1-7" data-line-number="7">## Or if we want to perform for all the matrix</a>
<a class="sourceLine" id="cb1-8" data-line-number="8">## A list of matrices to merge</a>
<a class="sourceLine" id="cb1-9" data-line-number="9">sim &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/list.html">list</a></span>(<span class="st">"kegg"</span> =<span class="st"> </span>kegg, <span class="st">"react"</span> =<span class="st"> </span>react)</a>
<a class="sourceLine" id="cb1-10" data-line-number="10"><span class="kw"><a href="../reference/similarities.html">similarities</a></span>(sim, weighted.sum, <span class="dt">w =</span> <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">0.3</span>, <span class="fl">0.7</span>))</a>
<a class="sourceLine" id="cb1-11" data-line-number="11">##            672       675         10</a>
<a class="sourceLine" id="cb1-12" data-line-number="12">## 672 1.00000000 0.7247289 0.09926055</a>
<a class="sourceLine" id="cb1-13" data-line-number="13">## 675 0.72472885 1.0000000 0.05211140</a>
<a class="sourceLine" id="cb1-14" data-line-number="14">## 10  0.09926055 0.0521114 1.00000000</a>
<a class="sourceLine" id="cb1-15" data-line-number="15"><span class="kw"><a href="../reference/similarities.html">similarities</a></span>(sim, weighted.prod, <span class="dt">w =</span> <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">0.3</span>, <span class="fl">0.7</span>))</a>
<a class="sourceLine" id="cb1-16" data-line-number="16">##           672          675           10</a>
<a class="sourceLine" id="cb1-17" data-line-number="17">## 672 0.2100000 0.0173101952 0.0000000000</a>
<a class="sourceLine" id="cb1-18" data-line-number="18">## 675 0.0173102 0.2100000000 0.0001227335</a>
<a class="sourceLine" id="cb1-19" data-line-number="19">## 10  0.0000000 0.0001227335 0.2100000000</a>
<a class="sourceLine" id="cb1-20" data-line-number="20"><span class="kw"><a href="../reference/similarities.html">similarities</a></span>(sim, prod)</a>
<a class="sourceLine" id="cb1-21" data-line-number="21">##           672          675           10</a>
<a class="sourceLine" id="cb1-22" data-line-number="22">## 672 1.0000000 0.0824295011 0.0000000000</a>
<a class="sourceLine" id="cb1-23" data-line-number="23">## 675 0.0824295 1.0000000000 0.0005844451</a>
<a class="sourceLine" id="cb1-24" data-line-number="24">## 10  0.0000000 0.0005844451 1.0000000000</a>
<a class="sourceLine" id="cb1-25" data-line-number="25"><span class="kw"><a href="../reference/similarities.html">similarities</a></span>(sim, mean)</a>
<a class="sourceLine" id="cb1-26" data-line-number="26">##            672        675         10</a>
<a class="sourceLine" id="cb1-27" data-line-number="27">## 672 1.00000000 0.54121475 0.07090039</a>
<a class="sourceLine" id="cb1-28" data-line-number="28">## 675 0.54121475 1.00000000 0.03957722</a>
<a class="sourceLine" id="cb1-29" data-line-number="29">## 10  0.07090039 0.03957722 1.00000000</a></code></pre></div>
<p>This functions are similar to <code>weighted.mean</code>, except that first the multiplication by the weights is done and then the <code>NA</code>s are removed:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb2-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/stats/weighted.mean.html">weighted.mean</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="dv">1</span>, <span class="ot">NA</span>), <span class="dt">w =</span> <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">0.5</span>, <span class="fl">0.5</span>), <span class="dt">na.rm =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb2-2" data-line-number="2">## [1] 1</a>
<a class="sourceLine" id="cb2-3" data-line-number="3"><span class="kw"><a href="https://rdrr.io/r/stats/weighted.mean.html">weighted.mean</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="dv">1</span>, <span class="fl">0.5</span>, <span class="ot">NA</span>), <span class="dt">w =</span> <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">0.5</span>, <span class="fl">0.25</span>, <span class="fl">0.25</span>), <span class="dt">na.rm =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb2-4" data-line-number="4">## [1] 0.8333333</a>
<a class="sourceLine" id="cb2-5" data-line-number="5"><span class="kw"><a href="../reference/weighted.html">weighted.sum</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="dv">1</span>, <span class="ot">NA</span>), <span class="dt">w =</span> <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">0.5</span>, <span class="fl">0.5</span>))</a>
<a class="sourceLine" id="cb2-6" data-line-number="6">## [1] 0.5</a>
<a class="sourceLine" id="cb2-7" data-line-number="7"><span class="kw"><a href="../reference/weighted.html">weighted.sum</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="dv">1</span>, <span class="fl">0.5</span>, <span class="ot">NA</span>), <span class="dt">w =</span> <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">0.5</span>, <span class="fl">0.25</span>, <span class="fl">0.25</span>))</a>
<a class="sourceLine" id="cb2-8" data-line-number="8">## [1] 0.625</a>
<a class="sourceLine" id="cb2-9" data-line-number="9"><span class="kw"><a href="../reference/weighted.html">weighted.prod</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="dv">1</span>, <span class="ot">NA</span>), <span class="dt">w =</span> <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">0.5</span>, <span class="fl">0.5</span>))</a>
<a class="sourceLine" id="cb2-10" data-line-number="10">## [1] 0.5</a>
<a class="sourceLine" id="cb2-11" data-line-number="11"><span class="kw"><a href="../reference/weighted.html">weighted.prod</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="dv">1</span>, <span class="fl">0.5</span>, <span class="ot">NA</span>), <span class="dt">w =</span> <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">0.5</span>, <span class="fl">0.25</span>, <span class="fl">0.25</span>))</a>
<a class="sourceLine" id="cb2-12" data-line-number="12">## [1] 0.0625</a></code></pre></div>
</div>
<div id="assessing-a-differential-study" class="section level1">
<h1 class="hasAnchor">
<a href="#assessing-a-differential-study" class="anchor"></a>Assessing a differential study</h1>
<p>In this example we will use the functional similarities in a classical differential study.</p>
<div id="obtaining-data" class="section level2">
<h2 class="hasAnchor">
<a href="#obtaining-data" class="anchor"></a>Obtaining data</h2>
<p>We start using data from the <a href="http://bioconductor.org/help/workflows/rnaseqGene/#differential-expression-analysis">RNAseq workflow</a> and following the analysis comparing treated and untreated:</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/base/message.html">suppressPackageStartupMessages</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">"airway"</span>))</a>
<a class="sourceLine" id="cb3-2" data-line-number="2"><span class="kw"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="st">"airway"</span>)</a>
<a class="sourceLine" id="cb3-3" data-line-number="3"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">"DESeq2"</span>)</a>
<a class="sourceLine" id="cb3-4" data-line-number="4"></a>
<a class="sourceLine" id="cb3-5" data-line-number="5">dds &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/DESeq2/man/DESeqDataSet.html">DESeqDataSet</a></span>(airway, <span class="dt">design =</span> <span class="op">~</span><span class="st"> </span>cell <span class="op">+</span><span class="st"> </span>dex)</a>
<a class="sourceLine" id="cb3-6" data-line-number="6">dds<span class="op">$</span>dex &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/stats/relevel.html">relevel</a></span>(dds<span class="op">$</span>dex, <span class="st">"untrt"</span>)</a>
<a class="sourceLine" id="cb3-7" data-line-number="7">dds &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/DESeq2/man/DESeq.html">DESeq</a></span>(dds)</a>
<a class="sourceLine" id="cb3-8" data-line-number="8">## estimating size factors</a>
<a class="sourceLine" id="cb3-9" data-line-number="9">## estimating dispersions</a>
<a class="sourceLine" id="cb3-10" data-line-number="10">## gene-wise dispersion estimates</a>
<a class="sourceLine" id="cb3-11" data-line-number="11">## mean-dispersion relationship</a>
<a class="sourceLine" id="cb3-12" data-line-number="12">## final dispersion estimates</a>
<a class="sourceLine" id="cb3-13" data-line-number="13">## fitting model and testing</a>
<a class="sourceLine" id="cb3-14" data-line-number="14">res &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/DESeq2/man/results.html">results</a></span>(dds, <span class="dt">alpha =</span> <span class="fl">0.05</span>)</a>
<a class="sourceLine" id="cb3-15" data-line-number="15"><span class="kw"><a href="https://rdrr.io/pkg/DESeq2/man/summary.html">summary</a></span>(res)</a>
<a class="sourceLine" id="cb3-16" data-line-number="16">## </a>
<a class="sourceLine" id="cb3-17" data-line-number="17">## out of 33469 with nonzero total read count</a>
<a class="sourceLine" id="cb3-18" data-line-number="18">## adjusted p-value &lt; 0.05</a>
<a class="sourceLine" id="cb3-19" data-line-number="19">## LFC &gt; 0 (up)       : 2211, 6.6%</a>
<a class="sourceLine" id="cb3-20" data-line-number="20">## LFC &lt; 0 (down)     : 1817, 5.4%</a>
<a class="sourceLine" id="cb3-21" data-line-number="21">## outliers [1]       : 0, 0%</a>
<a class="sourceLine" id="cb3-22" data-line-number="22">## low counts [2]     : 16676, 50%</a>
<a class="sourceLine" id="cb3-23" data-line-number="23">## (mean count &lt; 7)</a>
<a class="sourceLine" id="cb3-24" data-line-number="24">## [1] see 'cooksCutoff' argument of ?results</a>
<a class="sourceLine" id="cb3-25" data-line-number="25">## [2] see 'independentFiltering' argument of ?results</a>
<a class="sourceLine" id="cb3-26" data-line-number="26"><span class="kw"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(res<span class="op">$</span>log2FoldChange, <span class="op">-</span><span class="kw"><a href="https://rdrr.io/r/base/Log.html">log10</a></span>(res<span class="op">$</span>padj), <span class="dt">pch =</span> <span class="dv">16</span>, <span class="dt">xlab =</span> <span class="st">"log2FC"</span>, </a>
<a class="sourceLine" id="cb3-27" data-line-number="27">     <span class="dt">ylab =</span> <span class="st">"-log10(p.ajd)"</span>, <span class="dt">main =</span> <span class="st">"Untreated vs treated"</span>)</a>
<a class="sourceLine" id="cb3-28" data-line-number="28">logFC &lt;-<span class="st"> </span><span class="fl">2.5</span></a>
<a class="sourceLine" id="cb3-29" data-line-number="29"><span class="kw"><a href="https://rdrr.io/r/graphics/abline.html">abline</a></span>(<span class="dt">v=</span><span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="op">-</span>logFC, logFC), <span class="dt">h =</span> <span class="op">-</span><span class="kw"><a href="https://rdrr.io/r/base/Log.html">log10</a></span>(<span class="fl">0.05</span>), <span class="dt">col =</span> <span class="st">"red"</span>)</a></code></pre></div>
<div class="figure">
<img src="BioCor_2_advanced_files/figure-html/simulate-1.png" alt="Volcano plot. The airway data" width="700"><p class="caption">
Volcano plot. The airway data
</p>
</div>
<p>As we can see here there are around 4000 genes differentially expressed genes, some of them differentially expressed above 2^2.5.</p>
</div>
<div id="selecting-differentially-expressed-genes" class="section level2">
<h2 class="hasAnchor">
<a href="#selecting-differentially-expressed-genes" class="anchor"></a>Selecting differentially expressed genes</h2>
<p>Usually in such a study one selects genes above certain logFC or fold change threshold, here we use the absolute value of 2.5:</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" data-line-number="1">fc &lt;-<span class="st"> </span>res[<span class="kw"><a href="https://rdrr.io/r/base/MathFun.html">abs</a></span>(res<span class="op">$</span>log2FoldChange) <span class="op">&gt;=</span><span class="st"> </span>logFC <span class="op">&amp;</span><span class="st"> </span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span>(res<span class="op">$</span>padj), ]</a>
<a class="sourceLine" id="cb4-2" data-line-number="2">fc &lt;-<span class="st"> </span>fc[fc<span class="op">$</span>padj <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span>, ]</a>
<a class="sourceLine" id="cb4-3" data-line-number="3"><span class="co"># Convert Ids (used later)</span></a>
<a class="sourceLine" id="cb4-4" data-line-number="4">genes &lt;-<span class="st"> </span><span class="kw">select</span>(org.Hs.eg.db, <span class="dt">keys =</span> <span class="kw"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(res), <span class="dt">keytype =</span> <span class="st">"ENSEMBL"</span>, </a>
<a class="sourceLine" id="cb4-5" data-line-number="5">                <span class="dt">column =</span> <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"ENTREZID"</span>, <span class="st">"SYMBOL"</span>))</a>
<a class="sourceLine" id="cb4-6" data-line-number="6">## 'select()' returned 1:many mapping between keys and columns</a>
<a class="sourceLine" id="cb4-7" data-line-number="7">genesFC &lt;-<span class="st"> </span>genes[genes<span class="op">$</span>ENSEMBL <span class="op">%in%</span><span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(fc), ]</a>
<a class="sourceLine" id="cb4-8" data-line-number="8">genesFC &lt;-<span class="st"> </span>genesFC[<span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span>(genesFC<span class="op">$</span>ENTREZID), ]</a>
<a class="sourceLine" id="cb4-9" data-line-number="9">genesSim &lt;-<span class="st"> </span>genesFC<span class="op">$</span>ENTREZID</a>
<a class="sourceLine" id="cb4-10" data-line-number="10"><span class="kw"><a href="https://rdrr.io/r/base/names.html">names</a></span>(genesSim) &lt;-<span class="st"> </span>genesFC<span class="op">$</span>SYMBOL</a>
<a class="sourceLine" id="cb4-11" data-line-number="11">genesSim &lt;-<span class="st"> </span>genesSim[<span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/duplicated.html">duplicated</a></span>(genesSim)]</a>
<a class="sourceLine" id="cb4-12" data-line-number="12"><span class="co"># Calculate the functional similarity</span></a>
<a class="sourceLine" id="cb4-13" data-line-number="13">sim &lt;-<span class="st"> </span><span class="kw"><a href="../reference/mgeneSim.html">mgeneSim</a></span>(<span class="dt">genes =</span> genesSim, <span class="dt">info =</span> genesReact, <span class="dt">method =</span> <span class="st">"BMA"</span>)</a>
<a class="sourceLine" id="cb4-14" data-line-number="14">## Warning in mgeneSim(genes = genesSim, info = genesReact, method = "BMA"):</a>
<a class="sourceLine" id="cb4-15" data-line-number="15">## Some genes are not in the list provided.</a></code></pre></div>
<p>Once the similarities for the selected genes are calculated we can now visualize the effect of each method:</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" data-line-number="1">nas &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/apply.html">apply</a></span>(sim, <span class="dv">1</span>, <span class="cf">function</span>(x){<span class="kw"><a href="https://rdrr.io/r/base/all.html">all</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span>(x))})</a>
<a class="sourceLine" id="cb5-2" data-line-number="2">sim &lt;-<span class="st"> </span>sim[<span class="op">!</span>nas, <span class="op">!</span>nas]</a>
<a class="sourceLine" id="cb5-3" data-line-number="3"></a>
<a class="sourceLine" id="cb5-4" data-line-number="4">MDSs &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/stats/cmdscale.html">cmdscale</a></span>(<span class="dv">1</span><span class="op">-</span>sim)</a>
<a class="sourceLine" id="cb5-5" data-line-number="5"><span class="kw"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(MDSs, <span class="dt">type =</span> <span class="st">"n"</span>, <span class="dt">main =</span> <span class="st">"BMA similarity"</span>, <span class="dt">xlab =</span> <span class="st">"MDS1"</span>, <span class="dt">ylab =</span> <span class="st">"MDS2"</span>)</a>
<a class="sourceLine" id="cb5-6" data-line-number="6">up &lt;-<span class="st"> </span>genes[genes<span class="op">$</span>ENSEMBL <span class="op">%in%</span><span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(fc)[fc<span class="op">$</span>log2FoldChange <span class="op">&gt;=</span><span class="st"> </span>logFC], <span class="st">"SYMBOL"</span>]</a>
<a class="sourceLine" id="cb5-7" data-line-number="7"><span class="kw"><a href="https://rdrr.io/r/graphics/text.html">text</a></span>(MDSs, <span class="dt">labels =</span>  <span class="kw"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(MDSs), <span class="dt">col =</span> <span class="kw"><a href="https://rdrr.io/r/base/ifelse.html">ifelse</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(MDSs) <span class="op">%in%</span><span class="st"> </span>up, <span class="st">"black"</span>, <span class="st">"red"</span>))</a>
<a class="sourceLine" id="cb5-8" data-line-number="8"><span class="kw"><a href="https://rdrr.io/r/graphics/abline.html">abline</a></span>(<span class="dt">h =</span> <span class="dv">0</span>, <span class="dt">v =</span> <span class="dv">0</span>)</a>
<a class="sourceLine" id="cb5-9" data-line-number="9"><span class="kw"><a href="https://rdrr.io/r/graphics/legend.html">legend</a></span>(<span class="st">"top"</span>, <span class="dt">legend =</span> <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"Up-regulated"</span>, <span class="st">"Down-regulated"</span>), <span class="dt">fill =</span> <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"black"</span>, <span class="st">"red"</span>))</a></code></pre></div>
<div class="figure">
<img src="BioCor_2_advanced_files/figure-html/pval1-1.png" alt="Functional similarity of genes with logFC above 2,5. Similar genes cluster together." width="1440"><p class="caption">
Functional similarity of genes with logFC above 2,5. Similar genes cluster together.
</p>
</div>
<p>This plot illustrate that some differentially expressed genes are quite similar according to their pathways. Suggesting that there might be a relationship between them. Furthermore, some up-regulated genes seem functionally related to down-regulated genes indicating a precise regulation of the pathways where they are involved.</p>
<p>Note that here we are only using 72 genes from the original 127.</p>
</div>
<div id="are-differentially-expressed-genes-selected-by-their-functionality" class="section level2">
<h2 class="hasAnchor">
<a href="#are-differentially-expressed-genes-selected-by-their-functionality" class="anchor"></a>Are differentially expressed genes selected by their functionality?</h2>
<p>In the previous section we have seen that some differentially expressed genes are functionally related and that they have a high logFC value. Are genes differentially expressed more functional related than those which aren’t differential expressed?</p>
<p>For simplicity we will use a subset of 400 genes represented again in a volcano plot and we will look for the functional similarities between those genes:</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb6-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span>(<span class="dv">250</span>)</a>
<a class="sourceLine" id="cb6-2" data-line-number="2">subRes &lt;-<span class="st"> </span>res[<span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span>(res<span class="op">$</span>log2FoldChange), ]</a>
<a class="sourceLine" id="cb6-3" data-line-number="3">subs &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/sample.html">sample.int</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/nrow.html">nrow</a></span>(subRes), <span class="dt">size =</span> <span class="dv">400</span>)</a>
<a class="sourceLine" id="cb6-4" data-line-number="4">subRes &lt;-<span class="st"> </span>subRes[subs, ]</a>
<a class="sourceLine" id="cb6-5" data-line-number="5">g &lt;-<span class="st"> </span>genes[genes<span class="op">$</span>ENSEMBL <span class="op">%in%</span><span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(subRes), <span class="st">"ENTREZID"</span>]</a>
<a class="sourceLine" id="cb6-6" data-line-number="6">gS &lt;-<span class="st"> </span><span class="kw"><a href="../reference/mgeneSim.html">mgeneSim</a></span>(g[g <span class="op">%in%</span><span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/names.html">names</a></span>(genesReact)], genesReact, <span class="st">"BMA"</span>)</a>
<a class="sourceLine" id="cb6-7" data-line-number="7">deg &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(subRes[subRes<span class="op">$</span>padj <span class="op">&lt;</span><span class="st"> </span><span class="fl">0.05</span> <span class="op">&amp;</span><span class="st"> </span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span>(subRes<span class="op">$</span>padj), ])</a>
<a class="sourceLine" id="cb6-8" data-line-number="8">keep &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(gS) <span class="op">%in%</span><span class="st"> </span>genes[genes<span class="op">$</span>ENSEMBL <span class="op">%in%</span><span class="st"> </span>deg, <span class="st">"ENTREZID"</span>]</a>
<a class="sourceLine" id="cb6-9" data-line-number="9"><span class="kw"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(subRes<span class="op">$</span>log2FoldChange, <span class="op">-</span><span class="kw"><a href="https://rdrr.io/r/base/Log.html">log10</a></span>(subRes<span class="op">$</span>padj), <span class="dt">pch =</span> <span class="dv">16</span>, <span class="dt">xlab =</span> <span class="st">"log2FC"</span>, </a>
<a class="sourceLine" id="cb6-10" data-line-number="10">     <span class="dt">ylab =</span> <span class="st">"-log10(p.ajd)"</span>, <span class="dt">main =</span> <span class="st">"Untreated vs treated"</span>)</a>
<a class="sourceLine" id="cb6-11" data-line-number="11"><span class="kw"><a href="https://rdrr.io/r/graphics/abline.html">abline</a></span>(<span class="dt">v=</span><span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="op">-</span>logFC, logFC), <span class="dt">h =</span> <span class="op">-</span><span class="kw"><a href="https://rdrr.io/r/base/Log.html">log10</a></span>(<span class="fl">0.05</span>), <span class="dt">col =</span> <span class="st">"red"</span>)</a></code></pre></div>
<div class="figure">
<img src="BioCor_2_advanced_files/figure-html/setting-1.png" alt="Volcano plot of the subset of 400 genes. This subset will be used in the following sections" width="700"><p class="caption">
Volcano plot of the subset of 400 genes. This subset will be used in the following sections
</p>
</div>
<p>We can answer this by testing it empirically:</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">"boot"</span>)</a>
<a class="sourceLine" id="cb7-2" data-line-number="2"><span class="co"># The mean of genes differentially expressed</span></a>
<a class="sourceLine" id="cb7-3" data-line-number="3">(scoreDEG &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/mean.html">mean</a></span>(gS[<span class="op">!</span>keep, keep], <span class="dt">na.rm =</span> <span class="ot">TRUE</span>)) </a>
<a class="sourceLine" id="cb7-4" data-line-number="4">## [1] 0.1188639</a>
<a class="sourceLine" id="cb7-5" data-line-number="5">b &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/boot/man/boot.html">boot</a></span>(<span class="dt">data =</span> gS, <span class="dt">R =</span> <span class="dv">1000</span>, <span class="dt">statistic=</span> <span class="cf">function</span>(x, i){</a>
<a class="sourceLine" id="cb7-6" data-line-number="6">    g &lt;-<span class="st"> </span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(x) <span class="op">%in%</span><span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(x)[i]</a>
<a class="sourceLine" id="cb7-7" data-line-number="7">    <span class="kw"><a href="https://rdrr.io/r/base/mean.html">mean</a></span>(x[g, i], <span class="dt">na.rm =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb7-8" data-line-number="8">})</a>
<a class="sourceLine" id="cb7-9" data-line-number="9">(p.val &lt;-<span class="st"> </span>(<span class="dv">1</span> <span class="op">+</span><span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/sum.html">sum</a></span>(b<span class="op">$</span>t<span class="op">&gt;</span>scoreDEG))<span class="op">/</span><span class="dv">1001</span>)</a>
<a class="sourceLine" id="cb7-10" data-line-number="10">## [1] 0.9220779</a>
<a class="sourceLine" id="cb7-11" data-line-number="11"><span class="kw"><a href="https://rdrr.io/r/graphics/hist.html">hist</a></span>(b<span class="op">$</span>t, <span class="dt">main =</span> <span class="st">"Distribution of scores"</span>, <span class="dt">xlab =</span> <span class="st">"Similarity score"</span>)</a>
<a class="sourceLine" id="cb7-12" data-line-number="12"><span class="kw"><a href="https://rdrr.io/r/graphics/abline.html">abline</a></span>(<span class="dt">v =</span> scoreDEG, <span class="dt">col =</span> <span class="st">"red"</span>)</a></code></pre></div>
<div class="figure">
<img src="BioCor_2_advanced_files/figure-html/cluster2-1.png" alt="Distribution of scores between differentially expressed genes and those who aren't. The line indicates the mean score of the similarity between differentially expressed genes and those which aren't differentially expressed." width="700"><p class="caption">
Distribution of scores between differentially expressed genes and those who aren’t. The line indicates the mean score of the similarity between differentially expressed genes and those which aren’t differentially expressed.
</p>
</div>
<p>Comparing the genes differentially expressed and those who aren’t doesn’t show that they are non-randomly selected (p-value 0.9220779). The genes with a p-value below the threshold are not more closely functionally related than all the other genes<a href="#fn1" class="footnote-ref" id="fnref1"><sup>1</sup></a>.</p>
</div>
<div id="are-functionally-related-the-selected-differentially-expressed-genes" class="section level2">
<h2 class="hasAnchor">
<a href="#are-functionally-related-the-selected-differentially-expressed-genes" class="anchor"></a>Are functionally related the selected differentially expressed genes?</h2>
<p>We have seen that the genes differentially expressed aren’t selected by their functionality. However they could be more functionally related than the other genes. Are the differentially expressed genes more functionally similar than it would be expected ?</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb8-1" data-line-number="1">(scoreW &lt;-<span class="st"> </span><span class="kw"><a href="../reference/combineScores.html">combineScores</a></span>(gS[keep, keep], <span class="st">"avg"</span>))</a>
<a class="sourceLine" id="cb8-2" data-line-number="2">## [1] 0.1562942</a>
<a class="sourceLine" id="cb8-3" data-line-number="3">b &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/boot/man/boot.html">boot</a></span>(<span class="dt">data =</span> gS, <span class="dt">R =</span> <span class="dv">1000</span>, <span class="dt">statistic=</span> <span class="cf">function</span>(x, i){</a>
<a class="sourceLine" id="cb8-4" data-line-number="4">    <span class="kw"><a href="https://rdrr.io/r/base/mean.html">mean</a></span>(x[i, i], <span class="dt">na.rm =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb8-5" data-line-number="5">})</a>
<a class="sourceLine" id="cb8-6" data-line-number="6">(p.val &lt;-<span class="st"> </span>(<span class="dv">1</span> <span class="op">+</span><span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/sum.html">sum</a></span>(b<span class="op">$</span>t<span class="op">&gt;</span>scoreW))<span class="op">/</span><span class="dv">1001</span>) <span class="co"># P-value</span></a>
<a class="sourceLine" id="cb8-7" data-line-number="7">## [1] 0.01098901</a>
<a class="sourceLine" id="cb8-8" data-line-number="8"><span class="kw"><a href="https://rdrr.io/r/graphics/hist.html">hist</a></span>(b<span class="op">$</span>t, <span class="dt">main =</span> <span class="st">"Distribution of scores"</span>, <span class="dt">xlab =</span> <span class="st">"Similarity score"</span>)</a>
<a class="sourceLine" id="cb8-9" data-line-number="9"><span class="kw"><a href="https://rdrr.io/r/graphics/abline.html">abline</a></span>(<span class="dt">v =</span> scoreW, <span class="dt">col =</span> <span class="st">"red"</span>)</a></code></pre></div>
<div class="figure">
<img src="BioCor_2_advanced_files/figure-html/pval2-1.png" alt="Distribution of the similarity within differentially expressed genes. The line indicates the mean funtional similarity whitin them." width="700"><p class="caption">
Distribution of the similarity within differentially expressed genes. The line indicates the mean funtional similarity whitin them.
</p>
</div>
<p>If we selected randomly the genes from our pool we would expect a score around 0.1562942 with a probability of 0.010989. That means that the differentially expressed genes is highly different compared to the other genes if we use a significance threshold of 0.05 <a href="#fn2" class="footnote-ref" id="fnref2"><sup>2</sup></a>.</p>
</div>
<div id="influence-of-the-fold-change-in-the-functionally-similarity-of-the-genes" class="section level2">
<h2 class="hasAnchor">
<a href="#influence-of-the-fold-change-in-the-functionally-similarity-of-the-genes" class="anchor"></a>Influence of the fold change in the functionally similarity of the genes</h2>
<p>We have seen that the genes differentially expressed are not selected by functional similarity but they are functionally related. Now we would like to know if selecting a fold change threshold affects the functional similarity between them.</p>
<p>To know the relationship between the fold change and the similarity between genes we have several methods:</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" data-line-number="1">s &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/seq.html">seq</a></span>(<span class="dv">0</span>, <span class="kw"><a href="https://rdrr.io/r/base/Extremes.html">max</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/MathFun.html">abs</a></span>(subRes<span class="op">$</span>log2FoldChange))<span class="op">+</span><span class="fl">0.05</span>, <span class="dt">by =</span> <span class="fl">0.05</span>)</a>
<a class="sourceLine" id="cb9-2" data-line-number="2">l &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span>(s, <span class="cf">function</span>(x){</a>
<a class="sourceLine" id="cb9-3" data-line-number="3">    deg &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(subRes[<span class="kw"><a href="https://rdrr.io/r/base/MathFun.html">abs</a></span>(subRes<span class="op">$</span>log2FoldChange) <span class="op">&gt;=</span><span class="st"> </span>x, ])</a>
<a class="sourceLine" id="cb9-4" data-line-number="4">    keep &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(gS) <span class="op">%in%</span><span class="st"> </span>genes[genes<span class="op">$</span>ENSEMBL <span class="op">%in%</span><span class="st"> </span>deg, <span class="st">"ENTREZID"</span>]</a>
<a class="sourceLine" id="cb9-5" data-line-number="5">    BetweenAbove &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/mean.html">mean</a></span>(gS[keep, keep], <span class="dt">na.rm =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb9-6" data-line-number="6">    AboveBelow &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/mean.html">mean</a></span>(gS[keep, <span class="op">!</span>keep], <span class="dt">na.rm =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb9-7" data-line-number="7">    BetweenBelow &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/mean.html">mean</a></span>(gS[<span class="op">!</span>keep, <span class="op">!</span>keep], <span class="dt">na.rm =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb9-8" data-line-number="8">    <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"BetweenAbove"</span> =<span class="st"> </span>BetweenAbove, <span class="st">"AboveBelow"</span> =<span class="st"> </span>AboveBelow, </a>
<a class="sourceLine" id="cb9-9" data-line-number="9">      <span class="st">"BetweenBelow"</span> =<span class="st"> </span>BetweenBelow)</a>
<a class="sourceLine" id="cb9-10" data-line-number="10">})</a>
<a class="sourceLine" id="cb9-11" data-line-number="11">L &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/cbind.html">cbind</a></span>(<span class="dt">logfc =</span> s, <span class="kw"><a href="https://rdrr.io/r/base/t.html">t</a></span>(l)))</a>
<a class="sourceLine" id="cb9-12" data-line-number="12"><span class="kw"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(L<span class="op">$</span>logfc, L<span class="op">$</span>BetweenAbove, <span class="dt">type =</span> <span class="st">"l"</span>, <span class="dt">xlab =</span> <span class="st">"abs(log2) fold change"</span>, </a>
<a class="sourceLine" id="cb9-13" data-line-number="13">     <span class="dt">ylab =</span> <span class="st">"Similarity score"</span>, </a>
<a class="sourceLine" id="cb9-14" data-line-number="14">     <span class="dt">main =</span> <span class="st">"Similarity scores along logFC"</span>, <span class="dt">col =</span> <span class="st">"darkred"</span>)</a>
<a class="sourceLine" id="cb9-15" data-line-number="15"><span class="kw"><a href="https://rdrr.io/r/graphics/lines.html">lines</a></span>(L<span class="op">$</span>logfc, L<span class="op">$</span>AboveBelow, <span class="dt">col =</span> <span class="st">"darkgreen"</span>)</a>
<a class="sourceLine" id="cb9-16" data-line-number="16"><span class="kw"><a href="https://rdrr.io/r/graphics/lines.html">lines</a></span>(L<span class="op">$</span>logfc, L<span class="op">$</span>BetweenBelow, <span class="dt">col =</span> <span class="st">"black"</span>)</a>
<a class="sourceLine" id="cb9-17" data-line-number="17"><span class="kw"><a href="https://rdrr.io/r/graphics/legend.html">legend</a></span>(<span class="st">"topleft"</span>, </a>
<a class="sourceLine" id="cb9-18" data-line-number="18">       <span class="dt">legend =</span> <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"Between genes above and below threshold"</span>, </a>
<a class="sourceLine" id="cb9-19" data-line-number="19">                  <span class="st">"Whitin genes above threshold"</span>, </a>
<a class="sourceLine" id="cb9-20" data-line-number="20">                  <span class="st">"Whitin genes below threshold"</span>), </a>
<a class="sourceLine" id="cb9-21" data-line-number="21">       <span class="dt">fill =</span> <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"darkgreen"</span>, <span class="st">"darkred"</span>, <span class="st">"black"</span>))</a></code></pre></div>
<div class="figure">
<img src="BioCor_2_advanced_files/figure-html/logfc1-1.png" alt="Similarity of genes along abs(logFC). Assessing the similarity of genes according to their absolute log2 fold change." width="700"><p class="caption">
Similarity of genes along abs(logFC). Assessing the similarity of genes according to their absolute log2 fold change.
</p>
</div>
<p>The functional similarity of the genes above the threshold increases with a more restrictive threshold, indicating that a logFC threshold selects genes by their functionality, or in other words that genes differentially expressed tend to be of related pathways. The similarity between those genes above the threshold and below remains constant as well as within genes below the threshold.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb10-1" data-line-number="1">l &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span>(s, <span class="cf">function</span>(x){</a>
<a class="sourceLine" id="cb10-2" data-line-number="2">    <span class="co"># Names of genes up and down regulated</span></a>
<a class="sourceLine" id="cb10-3" data-line-number="3">    degUp &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(subRes[subRes<span class="op">$</span>log2FoldChange <span class="op">&gt;=</span><span class="st"> </span>x, ])</a>
<a class="sourceLine" id="cb10-4" data-line-number="4">    degDown &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(subRes[subRes<span class="op">$</span>log2FoldChange <span class="op">&lt;=</span><span class="st"> </span><span class="op">-</span>x, ])</a>
<a class="sourceLine" id="cb10-5" data-line-number="5">    </a>
<a class="sourceLine" id="cb10-6" data-line-number="6">    <span class="co"># Translate to ids in gS</span></a>
<a class="sourceLine" id="cb10-7" data-line-number="7">    keepUp &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(gS) <span class="op">%in%</span><span class="st"> </span>genes[genes<span class="op">$</span>ENSEMBL <span class="op">%in%</span><span class="st"> </span>degUp, <span class="st">"ENTREZID"</span>]</a>
<a class="sourceLine" id="cb10-8" data-line-number="8">    keepDown &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(gS) <span class="op">%in%</span><span class="st"> </span>genes[genes<span class="op">$</span>ENSEMBL <span class="op">%in%</span><span class="st"> </span>degDown, <span class="st">"ENTREZID"</span>]</a>
<a class="sourceLine" id="cb10-9" data-line-number="9">    </a>
<a class="sourceLine" id="cb10-10" data-line-number="10">    <span class="co"># Calculate the mean similarity between each subgrup</span></a>
<a class="sourceLine" id="cb10-11" data-line-number="11">    between &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/mean.html">mean</a></span>(gS[keepUp, keepDown], <span class="dt">na.rm =</span> <span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb10-12" data-line-number="12">    </a>
<a class="sourceLine" id="cb10-13" data-line-number="13">    <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"UpVsDown"</span> =<span class="st"> </span>between)</a>
<a class="sourceLine" id="cb10-14" data-line-number="14">})</a>
<a class="sourceLine" id="cb10-15" data-line-number="15">L &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/cbind.html">cbind</a></span>(<span class="st">"logfc"</span> =<span class="st"> </span>s, <span class="st">"UpVsDown"</span> =<span class="st"> </span>l))</a>
<a class="sourceLine" id="cb10-16" data-line-number="16"><span class="kw"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(L<span class="op">$</span>logfc, L<span class="op">$</span>UpVsDown, <span class="dt">type =</span> <span class="st">"l"</span>, <span class="dt">xlab =</span> <span class="st">"abs(log2) fold change threshold"</span>, <span class="dt">ylab =</span> <span class="st">"Similarity score"</span>, </a>
<a class="sourceLine" id="cb10-17" data-line-number="17">     <span class="dt">main =</span> <span class="st">"Similarity scores along logFC"</span>)</a>
<a class="sourceLine" id="cb10-18" data-line-number="18"><span class="kw"><a href="https://rdrr.io/r/graphics/legend.html">legend</a></span>(<span class="st">"topright"</span>, <span class="dt">legend =</span> <span class="st">"Up vs down regulated genes"</span>,</a>
<a class="sourceLine" id="cb10-19" data-line-number="19">       <span class="dt">fill =</span> <span class="st">"black"</span>)</a></code></pre></div>
<div class="figure">
<img src="BioCor_2_advanced_files/figure-html/logfc2-1.png" alt="Functional similarity between the up-regulated and down-regulated genes." width="700"><p class="caption">
Functional similarity between the up-regulated and down-regulated genes.
</p>
</div>
<p>The maximal functional similarity between genes up-regulated and down-regulated are at 1.15 log2 fold change.</p>
</div>
</div>
<div id="assessing-a-new-pathway" class="section level1">
<h1 class="hasAnchor">
<a href="#assessing-a-new-pathway" class="anchor"></a>Assessing a new pathway</h1>
<p>Sometimes the top differentially expressed genes or some other key genes are selected as a signature or a potential new group of related genes. In those cases we can test how does the network of genes change if we add them. Here we create a new pathway named <code>deg</code>, and we see the effect on the functional similarity score for all the genes:</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb11-1" data-line-number="1"><span class="co"># Adding a new pathway "deg" to those genes</span></a>
<a class="sourceLine" id="cb11-2" data-line-number="2">genesReact2 &lt;-<span class="st"> </span>genesReact</a>
<a class="sourceLine" id="cb11-3" data-line-number="3">diffGenes &lt;-<span class="st"> </span>genes[genes<span class="op">$</span>ENSEMBL <span class="op">%in%</span><span class="st"> </span>deg, <span class="st">"ENTREZID"</span>]</a>
<a class="sourceLine" id="cb11-4" data-line-number="4">genesReact2[diffGenes] &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span>(genesReact[diffGenes], <span class="cf">function</span>(x){<span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(x, <span class="st">"deg"</span>)})</a>
<a class="sourceLine" id="cb11-5" data-line-number="5"><span class="kw"><a href="https://rdrr.io/r/graphics/plot.html">plot</a></span>(<span class="kw"><a href="https://rdrr.io/r/stats/ecdf.html">ecdf</a></span>(<span class="kw"><a href="../reference/mgeneSim.html">mgeneSim</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/names.html">names</a></span>(genesReact), genesReact)))</a>
<a class="sourceLine" id="cb11-6" data-line-number="6"><span class="kw"><a href="https://rdrr.io/r/graphics/curve.html">curve</a></span>(<span class="kw"><a href="https://rdrr.io/r/stats/ecdf.html">ecdf</a></span>(<span class="kw"><a href="../reference/mgeneSim.html">mgeneSim</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/names.html">names</a></span>(genesReact2), genesReact2)), <span class="dt">color =</span> <span class="st">"red"</span>)</a></code></pre></div>
<p>This would take lot of time, for a illustration purpose we reduce to some genes:</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb12-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">"Hmisc"</span>)</a>
<a class="sourceLine" id="cb12-2" data-line-number="2">genesReact2 &lt;-<span class="st"> </span>genesReact</a>
<a class="sourceLine" id="cb12-3" data-line-number="3">diffGenes &lt;-<span class="st"> </span>genes[genes<span class="op">$</span>ENSEMBL <span class="op">%in%</span><span class="st"> </span>deg, <span class="st">"ENTREZID"</span>]</a>
<a class="sourceLine" id="cb12-4" data-line-number="4"><span class="co"># Create the new pathway called deg</span></a>
<a class="sourceLine" id="cb12-5" data-line-number="5">genesReact2[diffGenes] &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/lapply.html">sapply</a></span>(genesReact[diffGenes], <span class="cf">function</span>(x){<span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(x, <span class="st">"deg"</span>)})</a>
<a class="sourceLine" id="cb12-6" data-line-number="6">ids &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/unique.html">unique</a></span>(genes[genes<span class="op">$</span>ENSEMBL <span class="op">%in%</span><span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(subRes), <span class="st">"ENTREZID"</span>])</a>
<a class="sourceLine" id="cb12-7" data-line-number="7"><span class="kw"><a href="https://rdrr.io/pkg/Hmisc/man/Ecdf.html">Ecdf</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="kw"><a href="../reference/mgeneSim.html">mgeneSim</a></span>(ids, genesReact, <span class="dt">method =</span> <span class="st">"BMA"</span>),  </a>
<a class="sourceLine" id="cb12-8" data-line-number="8">       <span class="kw"><a href="../reference/mgeneSim.html">mgeneSim</a></span>(ids, genesReact2, <span class="dt">method =</span> <span class="st">"BMA"</span>)), </a>
<a class="sourceLine" id="cb12-9" data-line-number="9">     <span class="dt">group =</span> <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/rep.html">rep</a></span>(<span class="st">"Reactome"</span>, <span class="kw"><a href="https://rdrr.io/r/base/length.html">length</a></span>(ids)<span class="op">^</span><span class="dv">2</span>), <span class="kw"><a href="https://rdrr.io/r/base/rep.html">rep</a></span>(<span class="st">"Modified"</span>, <span class="kw"><a href="https://rdrr.io/r/base/length.html">length</a></span>(ids)<span class="op">^</span><span class="dv">2</span>)), </a>
<a class="sourceLine" id="cb12-10" data-line-number="10">     <span class="dt">col =</span> <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"black"</span>, <span class="st">"red"</span>), <span class="dt">xlab =</span> <span class="st">"Functional similarities"</span>, <span class="dt">main =</span> <span class="st">"Empirical cumulative distribution"</span>)</a></code></pre></div>
<div class="figure">
<img src="BioCor_2_advanced_files/figure-html/newPathway2-1.png" alt="The effect of adding a new pathway to a functional similarity. In red the same network as in black but with the added pathway." width="700"><p class="caption">
The effect of adding a new pathway to a functional similarity. In red the same network as in black but with the added pathway.
</p>
</div>
</div>
<div id="merging-sources-of-information" class="section level1">
<h1 class="hasAnchor">
<a href="#merging-sources-of-information" class="anchor"></a>Merging sources of information</h1>
<p>Sometimes we have several origin of information, either several databases, or information from other programs… We can merge this in the single object required by the function in <code>BioCor</code> using the function <code>combineSources</code><a href="#fn3" class="footnote-ref" id="fnref3"><sup>3</sup></a>.</p>
<p>This functions helps to evaluate what happens when we add more pathway information. For instance here we add the information in Kegg and the information in Reactome and we visualize it using the same procedure as previously:</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb13-1" data-line-number="1">genesKegg &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/list.html">as.list</a></span>(org.Hs.egPATH)</a>
<a class="sourceLine" id="cb13-2" data-line-number="2">gSK &lt;-<span class="st"> </span><span class="kw"><a href="../reference/mgeneSim.html">mgeneSim</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(gS), genesKegg)</a>
<a class="sourceLine" id="cb13-3" data-line-number="3">mix &lt;-<span class="st"> </span><span class="kw"><a href="../reference/combineSources.html">combineSources</a></span>(genesKegg, genesReact)</a>
<a class="sourceLine" id="cb13-4" data-line-number="4">## Warning in combineSources(genesKegg, genesReact): More than 50% of genes identifiers of a source are unique</a>
<a class="sourceLine" id="cb13-5" data-line-number="5">## Check the identifiers of the genes</a>
<a class="sourceLine" id="cb13-6" data-line-number="6">gSMix &lt;-<span class="st"> </span><span class="kw"><a href="../reference/mgeneSim.html">mgeneSim</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(gS), mix)</a>
<a class="sourceLine" id="cb13-7" data-line-number="7"><span class="kw"><a href="https://rdrr.io/pkg/Hmisc/man/Ecdf.html">Ecdf</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(gS, gSK, gSMix), </a>
<a class="sourceLine" id="cb13-8" data-line-number="8">     <span class="dt">group =</span> <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/rep.html">rep</a></span>(<span class="st">"Reactome"</span>, <span class="kw"><a href="https://rdrr.io/r/base/length.html">length</a></span>(gS)), <span class="kw"><a href="https://rdrr.io/r/base/rep.html">rep</a></span>(<span class="st">"Kegg"</span>, <span class="kw"><a href="https://rdrr.io/r/base/length.html">length</a></span>(gSK)), </a>
<a class="sourceLine" id="cb13-9" data-line-number="9">               <span class="kw"><a href="https://rdrr.io/r/base/rep.html">rep</a></span>(<span class="st">"Mix"</span>, <span class="kw"><a href="https://rdrr.io/r/base/length.html">length</a></span>(gSMix))), </a>
<a class="sourceLine" id="cb13-10" data-line-number="10">     <span class="dt">col =</span> <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"black"</span>, <span class="st">"red"</span>, <span class="st">"blue"</span>), <span class="dt">xlab =</span> <span class="st">"Functional similarities"</span>, <span class="dt">main =</span> <span class="st">"ecdf"</span>)</a>
<a class="sourceLine" id="cb13-11" data-line-number="11">## Warning in regularize.values(x, y, ties, missing(ties)): collapsing to</a>
<a class="sourceLine" id="cb13-12" data-line-number="12">## unique 'x' values</a>
<a class="sourceLine" id="cb13-13" data-line-number="13">## Warning in regularize.values(x, y, ties, missing(ties)): collapsing to</a>
<a class="sourceLine" id="cb13-14" data-line-number="14">## unique 'x' values</a>
<a class="sourceLine" id="cb13-15" data-line-number="15"></a>
<a class="sourceLine" id="cb13-16" data-line-number="16">## Warning in regularize.values(x, y, ties, missing(ties)): collapsing to</a>
<a class="sourceLine" id="cb13-17" data-line-number="17">## unique 'x' values</a></code></pre></div>
<div class="figure">
<img src="BioCor_2_advanced_files/figure-html/combineSource-1.png" alt="Comparison of functional similarity in different databases." width="700"><p class="caption">
Comparison of functional similarity in different databases.
</p>
</div>
<p>When mixed, there is a huge increase in the genes that share a pathway using the <code>max</code> method. Observe in next figure (@ref(fig:combineSource2)) how does the method affects to the results:</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb14-1" data-line-number="1">gSK2 &lt;-<span class="st"> </span><span class="kw"><a href="../reference/mgeneSim.html">mgeneSim</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(gS), genesKegg, <span class="dt">method =</span> <span class="st">"BMA"</span>)</a>
<a class="sourceLine" id="cb14-2" data-line-number="2">gS2 &lt;-<span class="st"> </span><span class="kw"><a href="../reference/mgeneSim.html">mgeneSim</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(gS), genesReact, <span class="dt">method =</span> <span class="st">"BMA"</span>)</a>
<a class="sourceLine" id="cb14-3" data-line-number="3">gSMix2 &lt;-<span class="st"> </span><span class="kw"><a href="../reference/mgeneSim.html">mgeneSim</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(gS), mix, <span class="dt">method =</span> <span class="st">"BMA"</span>)</a>
<a class="sourceLine" id="cb14-4" data-line-number="4"><span class="kw"><a href="https://rdrr.io/pkg/Hmisc/man/Ecdf.html">Ecdf</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(gS2, gSK2, gSMix2), </a>
<a class="sourceLine" id="cb14-5" data-line-number="5">     <span class="dt">group =</span> <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/rep.html">rep</a></span>(<span class="st">"Reactome"</span>, <span class="kw"><a href="https://rdrr.io/r/base/length.html">length</a></span>(gS)), <span class="kw"><a href="https://rdrr.io/r/base/rep.html">rep</a></span>(<span class="st">"Kegg"</span>, <span class="kw"><a href="https://rdrr.io/r/base/length.html">length</a></span>(gSK)), </a>
<a class="sourceLine" id="cb14-6" data-line-number="6">               <span class="kw"><a href="https://rdrr.io/r/base/rep.html">rep</a></span>(<span class="st">"Mix"</span>, <span class="kw"><a href="https://rdrr.io/r/base/length.html">length</a></span>(gSMix))), </a>
<a class="sourceLine" id="cb14-7" data-line-number="7">     <span class="dt">col =</span> <span class="kw"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"black"</span>, <span class="st">"red"</span>, <span class="st">"blue"</span>), <span class="dt">xlab =</span> <span class="st">"Functional similarities (BMA)"</span>, <span class="dt">main =</span> <span class="st">"ecdf"</span>)</a>
<a class="sourceLine" id="cb14-8" data-line-number="8">## Warning in regularize.values(x, y, ties, missing(ties)): collapsing to</a>
<a class="sourceLine" id="cb14-9" data-line-number="9">## unique 'x' values</a>
<a class="sourceLine" id="cb14-10" data-line-number="10"></a>
<a class="sourceLine" id="cb14-11" data-line-number="11">## Warning in regularize.values(x, y, ties, missing(ties)): collapsing to</a>
<a class="sourceLine" id="cb14-12" data-line-number="12">## unique 'x' values</a>
<a class="sourceLine" id="cb14-13" data-line-number="13"></a>
<a class="sourceLine" id="cb14-14" data-line-number="14">## Warning in regularize.values(x, y, ties, missing(ties)): collapsing to</a>
<a class="sourceLine" id="cb14-15" data-line-number="15">## unique 'x' values</a></code></pre></div>
<div class="figure">
<img src="BioCor_2_advanced_files/figure-html/combineSource2-1.png" alt="Comparison of functional similarity in different gene sets." width="700"><p class="caption">
Comparison of functional similarity in different gene sets.
</p>
</div>
<p>Now we can appreciate that most of the functional similarity is brought by Reactome database</p>
</div>
<div id="mirna-analysis" class="section level1">
<h1 class="hasAnchor">
<a href="#mirna-analysis" class="anchor"></a>miRNA analysis</h1>
<p>miRNAs are RNAs that interact with many genes and transcripts and is subject to change with location and time, thus defining the effect of an miRNA is difficult. In this section we try to answer how functionally similar are miRNA between them to provide a help for potentially closely related miRNA.</p>
<p>First we look for human miRNAs and prepare them for using them as the input for the cluster functions (Restricting to 10 miRNA with less than 150 genes).<a href="#fn4" class="footnote-ref" id="fnref4"><sup>4</sup></a></p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb15-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">"targetscan.Hs.eg.db"</span>)</a>
<a class="sourceLine" id="cb15-2" data-line-number="2">## select human mirna </a>
<a class="sourceLine" id="cb15-3" data-line-number="3">humanMirnaIdx &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/grep.html">grep</a></span>(<span class="st">"hsa"</span>, <span class="kw">mappedkeys</span>(targetscan.Hs.egMIRNA)) </a>
<a class="sourceLine" id="cb15-4" data-line-number="4">## select seed-based families for human mirna</a>
<a class="sourceLine" id="cb15-5" data-line-number="5">humanMirna &lt;-<span class="st"> </span><span class="kw">mappedkeys</span>(targetscan.Hs.egMIRNA)[humanMirnaIdx]</a>
<a class="sourceLine" id="cb15-6" data-line-number="6">## select targets of families </a>
<a class="sourceLine" id="cb15-7" data-line-number="7">humanMirnaFamilies &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/unlist.html">unlist</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/get.html">mget</a></span>(humanMirna, targetscan.Hs.egMIRBASE2FAMILY)) </a>
<a class="sourceLine" id="cb15-8" data-line-number="8">humanMirnaTargets &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/get.html">mget</a></span>(humanMirnaFamilies, <span class="kw">revmap</span>(targetscan.Hs.egTARGETS))</a>
<a class="sourceLine" id="cb15-9" data-line-number="9"><span class="kw"><a href="https://rdrr.io/r/base/names.html">names</a></span>(humanMirnaTargets) &lt;-<span class="st"> </span>humanMirna</a>
<a class="sourceLine" id="cb15-10" data-line-number="10"><span class="co"># Restrict to miRNA with more than one target and less than 150</span></a>
<a class="sourceLine" id="cb15-11" data-line-number="11">miRNAs &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/sample.html">sample</a></span>(humanMirnaTargets[<span class="kw"><a href="https://rdrr.io/r/base/lengths.html">lengths</a></span>(humanMirnaTargets) <span class="op">&gt;</span><span class="st"> </span><span class="dv">1</span> <span class="op">&amp;</span></a>
<a class="sourceLine" id="cb15-12" data-line-number="12"><span class="st">                                       </span><span class="kw"><a href="https://rdrr.io/r/base/lengths.html">lengths</a></span>(humanMirnaTargets) <span class="op">&lt;</span><span class="st"> </span><span class="dv">150</span>], <span class="dv">10</span>)</a>
<a class="sourceLine" id="cb15-13" data-line-number="13"><span class="kw"><a href="https://rdrr.io/r/base/lengths.html">lengths</a></span>(miRNAs)</a>
<a class="sourceLine" id="cb15-14" data-line-number="14">##    hsa-miR-187 hsa-miR-127-3p   hsa-miR-551a    hsa-miR-191 hsa-miR-615-3p </a>
<a class="sourceLine" id="cb15-15" data-line-number="15">##             13             17             17            118             28 </a>
<a class="sourceLine" id="cb15-16" data-line-number="16">##    hsa-miR-99b   hsa-miR-551b   hsa-miR-487b    hsa-miR-184    hsa-miR-451 </a>
<a class="sourceLine" id="cb15-17" data-line-number="17">##             92             17             23             54             36</a></code></pre></div>
<p>Now we calculate the functional similarity of those miRNAs using a cluster approach.</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb16-1" data-line-number="1">cluster1 &lt;-<span class="st"> </span><span class="kw"><a href="../reference/mclusterSim.html">mclusterSim</a></span>(miRNAs, genesReact, <span class="dt">method =</span> <span class="st">"BMA"</span>)</a>
<a class="sourceLine" id="cb16-2" data-line-number="2">## Warning in mclusterSim(miRNAs, genesReact, method = "BMA"): Some genes are</a>
<a class="sourceLine" id="cb16-3" data-line-number="3">## not in the list provided.</a>
<a class="sourceLine" id="cb16-4" data-line-number="4">knitr<span class="op">::</span><span class="kw"><a href="https://rdrr.io/pkg/knitr/man/kable.html">kable</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/Round.html">round</a></span>(cluster1, <span class="dv">2</span>), <span class="dt">caption =</span> <span class="st">"The similarity between miRNA"</span>, <span class="dt">format =</span> <span class="st">"html"</span>)</a></code></pre></div>
<table class="table">
<caption>
The similarity between miRNA
</caption>
<thead><tr>
<th style="text-align:left;">
</th>
<th style="text-align:right;">
hsa-miR-187
</th>
<th style="text-align:right;">
hsa-miR-127-3p
</th>
<th style="text-align:right;">
hsa-miR-551a
</th>
<th style="text-align:right;">
hsa-miR-191
</th>
<th style="text-align:right;">
hsa-miR-615-3p
</th>
<th style="text-align:right;">
hsa-miR-99b
</th>
<th style="text-align:right;">
hsa-miR-551b
</th>
<th style="text-align:right;">
hsa-miR-487b
</th>
<th style="text-align:right;">
hsa-miR-184
</th>
<th style="text-align:right;">
hsa-miR-451
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
hsa-miR-187
</td>
<td style="text-align:right;">
1.00
</td>
<td style="text-align:right;">
0.55
</td>
<td style="text-align:right;">
0.58
</td>
<td style="text-align:right;">
0.61
</td>
<td style="text-align:right;">
0.55
</td>
<td style="text-align:right;">
0.69
</td>
<td style="text-align:right;">
0.58
</td>
<td style="text-align:right;">
0.62
</td>
<td style="text-align:right;">
0.68
</td>
<td style="text-align:right;">
0.49
</td>
</tr>
<tr>
<td style="text-align:left;">
hsa-miR-127-3p
</td>
<td style="text-align:right;">
0.55
</td>
<td style="text-align:right;">
1.00
</td>
<td style="text-align:right;">
0.46
</td>
<td style="text-align:right;">
0.49
</td>
<td style="text-align:right;">
0.47
</td>
<td style="text-align:right;">
0.48
</td>
<td style="text-align:right;">
0.46
</td>
<td style="text-align:right;">
0.49
</td>
<td style="text-align:right;">
0.45
</td>
<td style="text-align:right;">
0.44
</td>
</tr>
<tr>
<td style="text-align:left;">
hsa-miR-551a
</td>
<td style="text-align:right;">
0.58
</td>
<td style="text-align:right;">
0.46
</td>
<td style="text-align:right;">
1.00
</td>
<td style="text-align:right;">
0.68
</td>
<td style="text-align:right;">
0.82
</td>
<td style="text-align:right;">
0.62
</td>
<td style="text-align:right;">
1.00
</td>
<td style="text-align:right;">
0.78
</td>
<td style="text-align:right;">
0.65
</td>
<td style="text-align:right;">
0.55
</td>
</tr>
<tr>
<td style="text-align:left;">
hsa-miR-191
</td>
<td style="text-align:right;">
0.61
</td>
<td style="text-align:right;">
0.49
</td>
<td style="text-align:right;">
0.68
</td>
<td style="text-align:right;">
1.00
</td>
<td style="text-align:right;">
0.72
</td>
<td style="text-align:right;">
0.80
</td>
<td style="text-align:right;">
0.68
</td>
<td style="text-align:right;">
0.73
</td>
<td style="text-align:right;">
0.73
</td>
<td style="text-align:right;">
0.69
</td>
</tr>
<tr>
<td style="text-align:left;">
hsa-miR-615-3p
</td>
<td style="text-align:right;">
0.55
</td>
<td style="text-align:right;">
0.47
</td>
<td style="text-align:right;">
0.82
</td>
<td style="text-align:right;">
0.72
</td>
<td style="text-align:right;">
1.00
</td>
<td style="text-align:right;">
0.66
</td>
<td style="text-align:right;">
0.82
</td>
<td style="text-align:right;">
0.79
</td>
<td style="text-align:right;">
0.61
</td>
<td style="text-align:right;">
0.59
</td>
</tr>
<tr>
<td style="text-align:left;">
hsa-miR-99b
</td>
<td style="text-align:right;">
0.69
</td>
<td style="text-align:right;">
0.48
</td>
<td style="text-align:right;">
0.62
</td>
<td style="text-align:right;">
0.80
</td>
<td style="text-align:right;">
0.66
</td>
<td style="text-align:right;">
1.00
</td>
<td style="text-align:right;">
0.62
</td>
<td style="text-align:right;">
0.73
</td>
<td style="text-align:right;">
0.75
</td>
<td style="text-align:right;">
0.71
</td>
</tr>
<tr>
<td style="text-align:left;">
hsa-miR-551b
</td>
<td style="text-align:right;">
0.58
</td>
<td style="text-align:right;">
0.46
</td>
<td style="text-align:right;">
1.00
</td>
<td style="text-align:right;">
0.68
</td>
<td style="text-align:right;">
0.82
</td>
<td style="text-align:right;">
0.62
</td>
<td style="text-align:right;">
1.00
</td>
<td style="text-align:right;">
0.78
</td>
<td style="text-align:right;">
0.65
</td>
<td style="text-align:right;">
0.55
</td>
</tr>
<tr>
<td style="text-align:left;">
hsa-miR-487b
</td>
<td style="text-align:right;">
0.62
</td>
<td style="text-align:right;">
0.49
</td>
<td style="text-align:right;">
0.78
</td>
<td style="text-align:right;">
0.73
</td>
<td style="text-align:right;">
0.79
</td>
<td style="text-align:right;">
0.73
</td>
<td style="text-align:right;">
0.78
</td>
<td style="text-align:right;">
1.00
</td>
<td style="text-align:right;">
0.66
</td>
<td style="text-align:right;">
0.66
</td>
</tr>
<tr>
<td style="text-align:left;">
hsa-miR-184
</td>
<td style="text-align:right;">
0.68
</td>
<td style="text-align:right;">
0.45
</td>
<td style="text-align:right;">
0.65
</td>
<td style="text-align:right;">
0.73
</td>
<td style="text-align:right;">
0.61
</td>
<td style="text-align:right;">
0.75
</td>
<td style="text-align:right;">
0.65
</td>
<td style="text-align:right;">
0.66
</td>
<td style="text-align:right;">
1.00
</td>
<td style="text-align:right;">
0.67
</td>
</tr>
<tr>
<td style="text-align:left;">
hsa-miR-451
</td>
<td style="text-align:right;">
0.49
</td>
<td style="text-align:right;">
0.44
</td>
<td style="text-align:right;">
0.55
</td>
<td style="text-align:right;">
0.69
</td>
<td style="text-align:right;">
0.59
</td>
<td style="text-align:right;">
0.71
</td>
<td style="text-align:right;">
0.55
</td>
<td style="text-align:right;">
0.66
</td>
<td style="text-align:right;">
0.67
</td>
<td style="text-align:right;">
1.00
</td>
</tr>
</tbody>
</table>
<p>So for instance hsa-miR-551b, hsa-miR-551a are functionally related despite being from different families.</p>
</div>
<div id="comparing-with-go-similarities" class="section level1">
<h1 class="hasAnchor">
<a href="#comparing-with-go-similarities" class="anchor"></a>Comparing with GO similarities</h1>
<p>As suggested in the main vignette functional similarities can be compared to semantic similarities such as those based on GO. Here a comparison using the biological process from the gene ontologies is done:</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb17-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="st">"GOSemSim"</span>)</a>
<a class="sourceLine" id="cb17-2" data-line-number="2">## </a>
<a class="sourceLine" id="cb17-3" data-line-number="3">## GOSemSim v2.11.0  For help: https://guangchuangyu.github.io/GOSemSim</a>
<a class="sourceLine" id="cb17-4" data-line-number="4">## </a>
<a class="sourceLine" id="cb17-5" data-line-number="5">## If you use GOSemSim in published research, please cite:</a>
<a class="sourceLine" id="cb17-6" data-line-number="6">## Guangchuang Yu, Fei Li, Yide Qin, Xiaochen Bo, Yibo Wu, Shengqi Wang. GOSemSim: an R package for measuring semantic similarity among GO terms and gene products Bioinformatics 2010, 26(7):976-978</a>
<a class="sourceLine" id="cb17-7" data-line-number="7">## </a>
<a class="sourceLine" id="cb17-8" data-line-number="8">## Attaching package: 'GOSemSim'</a>
<a class="sourceLine" id="cb17-9" data-line-number="9">## The following objects are masked from 'package:BioCor':</a>
<a class="sourceLine" id="cb17-10" data-line-number="10">## </a>
<a class="sourceLine" id="cb17-11" data-line-number="11">##     clusterSim, combineScores, geneSim, mclusterSim, mgeneSim</a>
<a class="sourceLine" id="cb17-12" data-line-number="12">BP &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/pkg/GOSemSim/man/godata.html">godata</a></span>(<span class="st">'org.Hs.eg.db'</span>, <span class="dt">ont=</span><span class="st">"BP"</span>, <span class="dt">computeIC=</span><span class="ot">TRUE</span>)</a>
<a class="sourceLine" id="cb17-13" data-line-number="13">## preparing gene to GO mapping data...</a>
<a class="sourceLine" id="cb17-14" data-line-number="14">## preparing IC data...</a>
<a class="sourceLine" id="cb17-15" data-line-number="15">gsGO &lt;-<span class="st"> </span>GOSemSim<span class="op">::</span><span class="kw"><a href="https://rdrr.io/pkg/GOSemSim/man/mgeneSim.html">mgeneSim</a></span>(<span class="kw"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(gS), <span class="dt">semData =</span> BP, <span class="dt">measure =</span> <span class="st">"Resnik"</span>, <span class="dt">verbose =</span> <span class="ot">FALSE</span>)</a>
<a class="sourceLine" id="cb17-16" data-line-number="16">keep &lt;-<span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(gS) <span class="op">%in%</span><span class="st"> </span><span class="kw"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(gsGO)</a>
<a class="sourceLine" id="cb17-17" data-line-number="17"><span class="kw"><a href="https://rdrr.io/r/graphics/hist.html">hist</a></span>(<span class="kw"><a href="https://rdrr.io/r/stats/dist.html">as.dist</a></span>(gS[keep, keep]<span class="op">-</span>gsGO), </a>
<a class="sourceLine" id="cb17-18" data-line-number="18">     <span class="dt">main =</span> <span class="st">"Difference between functional similarity and biological process"</span>, </a>
<a class="sourceLine" id="cb17-19" data-line-number="19">     <span class="dt">xlab =</span> <span class="st">"Functional similarity - biological process similarity"</span>)</a></code></pre></div>
<div class="figure">
<img src="BioCor_2_advanced_files/figure-html/GOSemSim-1.png" alt="Comparison of similarities. Functional similarities compared to biological process semantic similarity." width="700"><p class="caption">
Comparison of similarities. Functional similarities compared to biological process semantic similarity.
</p>
</div>
<p>On this graphic we can observe that some genes have a large functional similarity and few biological similarity. They are present together in several pathways while they share few biological process, indicating that they might be key elements of the pathways they are in. On the other hand some other pairs of genes show higher biological process similarity than functional similarity, indicating specialization or compartmentalization of said genes.</p>
</div>
<div id="session-info" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#session-info" class="anchor"></a>Session Info</h1>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb18-1" data-line-number="1"><span class="kw"><a href="https://rdrr.io/r/utils/sessionInfo.html">sessionInfo</a></span>()</a>
<a class="sourceLine" id="cb18-2" data-line-number="2">## R version 3.6.1 (2017-01-27)</a>
<a class="sourceLine" id="cb18-3" data-line-number="3">## Platform: x86_64-pc-linux-gnu (64-bit)</a>
<a class="sourceLine" id="cb18-4" data-line-number="4">## Running under: Ubuntu 16.04.6 LTS</a>
<a class="sourceLine" id="cb18-5" data-line-number="5">## </a>
<a class="sourceLine" id="cb18-6" data-line-number="6">## Matrix products: default</a>
<a class="sourceLine" id="cb18-7" data-line-number="7">## BLAS:   /home/travis/R-bin/lib/R/lib/libRblas.so</a>
<a class="sourceLine" id="cb18-8" data-line-number="8">## LAPACK: /home/travis/R-bin/lib/R/lib/libRlapack.so</a>
<a class="sourceLine" id="cb18-9" data-line-number="9">## </a>
<a class="sourceLine" id="cb18-10" data-line-number="10">## locale:</a>
<a class="sourceLine" id="cb18-11" data-line-number="11">##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </a>
<a class="sourceLine" id="cb18-12" data-line-number="12">##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </a>
<a class="sourceLine" id="cb18-13" data-line-number="13">##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </a>
<a class="sourceLine" id="cb18-14" data-line-number="14">##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </a>
<a class="sourceLine" id="cb18-15" data-line-number="15">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </a>
<a class="sourceLine" id="cb18-16" data-line-number="16">## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </a>
<a class="sourceLine" id="cb18-17" data-line-number="17">## </a>
<a class="sourceLine" id="cb18-18" data-line-number="18">## attached base packages:</a>
<a class="sourceLine" id="cb18-19" data-line-number="19">## [1] parallel  stats4    stats     graphics  grDevices utils     datasets </a>
<a class="sourceLine" id="cb18-20" data-line-number="20">## [8] methods   base     </a>
<a class="sourceLine" id="cb18-21" data-line-number="21">## </a>
<a class="sourceLine" id="cb18-22" data-line-number="22">## other attached packages:</a>
<a class="sourceLine" id="cb18-23" data-line-number="23">##  [1] GOSemSim_2.11.0             targetscan.Hs.eg.db_0.6.1  </a>
<a class="sourceLine" id="cb18-24" data-line-number="24">##  [3] Hmisc_4.2-0                 ggplot2_3.2.1              </a>
<a class="sourceLine" id="cb18-25" data-line-number="25">##  [5] Formula_1.2-3               survival_2.44-1.1          </a>
<a class="sourceLine" id="cb18-26" data-line-number="26">##  [7] lattice_0.20-38             boot_1.3-22                </a>
<a class="sourceLine" id="cb18-27" data-line-number="27">##  [9] DESeq2_1.25.13              airway_1.5.0               </a>
<a class="sourceLine" id="cb18-28" data-line-number="28">## [11] SummarizedExperiment_1.15.9 DelayedArray_0.11.6        </a>
<a class="sourceLine" id="cb18-29" data-line-number="29">## [13] BiocParallel_1.19.3         matrixStats_0.55.0         </a>
<a class="sourceLine" id="cb18-30" data-line-number="30">## [15] GenomicRanges_1.37.16       GenomeInfoDb_1.21.2        </a>
<a class="sourceLine" id="cb18-31" data-line-number="31">## [17] reactome.db_1.68.0          org.Hs.eg.db_3.8.2         </a>
<a class="sourceLine" id="cb18-32" data-line-number="32">## [19] AnnotationDbi_1.47.1        IRanges_2.19.16            </a>
<a class="sourceLine" id="cb18-33" data-line-number="33">## [21] S4Vectors_0.23.25           Biobase_2.45.1             </a>
<a class="sourceLine" id="cb18-34" data-line-number="34">## [23] BiocGenerics_0.31.6         BioCor_1.9.1               </a>
<a class="sourceLine" id="cb18-35" data-line-number="35">## [25] BiocStyle_2.13.2           </a>
<a class="sourceLine" id="cb18-36" data-line-number="36">## </a>
<a class="sourceLine" id="cb18-37" data-line-number="37">## loaded via a namespace (and not attached):</a>
<a class="sourceLine" id="cb18-38" data-line-number="38">##  [1] bitops_1.0-6           fs_1.3.1               bit64_0.9-7           </a>
<a class="sourceLine" id="cb18-39" data-line-number="39">##  [4] RColorBrewer_1.1-2     rprojroot_1.3-2        tools_3.6.1           </a>
<a class="sourceLine" id="cb18-40" data-line-number="40">##  [7] backports_1.1.5        R6_2.4.0               rpart_4.1-15          </a>
<a class="sourceLine" id="cb18-41" data-line-number="41">## [10] DBI_1.0.0              lazyeval_0.2.2         colorspace_1.4-1      </a>
<a class="sourceLine" id="cb18-42" data-line-number="42">## [13] nnet_7.3-12            withr_2.1.2            gridExtra_2.3         </a>
<a class="sourceLine" id="cb18-43" data-line-number="43">## [16] bit_1.1-14             compiler_3.6.1         graph_1.63.0          </a>
<a class="sourceLine" id="cb18-44" data-line-number="44">## [19] htmlTable_1.13.2       desc_1.2.0             bookdown_0.14         </a>
<a class="sourceLine" id="cb18-45" data-line-number="45">## [22] scales_1.0.0           checkmate_1.9.4        genefilter_1.67.1     </a>
<a class="sourceLine" id="cb18-46" data-line-number="46">## [25] pkgdown_1.4.1          stringr_1.4.0          digest_0.6.21         </a>
<a class="sourceLine" id="cb18-47" data-line-number="47">## [28] foreign_0.8-71         rmarkdown_1.16         XVector_0.25.0        </a>
<a class="sourceLine" id="cb18-48" data-line-number="48">## [31] base64enc_0.1-3        pkgconfig_2.0.3        htmltools_0.3.6       </a>
<a class="sourceLine" id="cb18-49" data-line-number="49">## [34] highr_0.8              htmlwidgets_1.3        rlang_0.4.0           </a>
<a class="sourceLine" id="cb18-50" data-line-number="50">## [37] rstudioapi_0.10        RSQLite_2.1.2          acepack_1.4.1         </a>
<a class="sourceLine" id="cb18-51" data-line-number="51">## [40] RCurl_1.95-4.12        magrittr_1.5           GO.db_3.8.2           </a>
<a class="sourceLine" id="cb18-52" data-line-number="52">## [43] GenomeInfoDbData_1.2.1 Matrix_1.2-17          Rcpp_1.0.2            </a>
<a class="sourceLine" id="cb18-53" data-line-number="53">## [46] munsell_0.5.0          stringi_1.4.3          yaml_2.2.0            </a>
<a class="sourceLine" id="cb18-54" data-line-number="54">## [49] MASS_7.3-51.4          zlibbioc_1.31.0        grid_3.6.1            </a>
<a class="sourceLine" id="cb18-55" data-line-number="55">## [52] blob_1.2.0             crayon_1.3.4           splines_3.6.1         </a>
<a class="sourceLine" id="cb18-56" data-line-number="56">## [55] annotate_1.63.0        locfit_1.5-9.1         zeallot_0.1.0         </a>
<a class="sourceLine" id="cb18-57" data-line-number="57">## [58] knitr_1.25             pillar_1.4.2           geneplotter_1.63.0    </a>
<a class="sourceLine" id="cb18-58" data-line-number="58">## [61] codetools_0.2-16       XML_3.98-1.20          evaluate_0.14         </a>
<a class="sourceLine" id="cb18-59" data-line-number="59">## [64] latticeExtra_0.6-28    data.table_1.12.4      BiocManager_1.30.4    </a>
<a class="sourceLine" id="cb18-60" data-line-number="60">## [67] vctrs_0.2.0            gtable_0.3.0           assertthat_0.2.1      </a>
<a class="sourceLine" id="cb18-61" data-line-number="61">## [70] xfun_0.10              xtable_1.8-4           tibble_2.1.3          </a>
<a class="sourceLine" id="cb18-62" data-line-number="62">## [73] memoise_1.1.0          cluster_2.1.0          GSEABase_1.47.0</a></code></pre></div>
</div>
<div class="footnotes">
<hr>
<ol>
<li id="fn1"><p>From 400 genes there are 123 with pathway information and only 23 where significantly differentially expressed in this subset.<a href="#fnref1" class="footnote-back">↩</a></p></li>
<li id="fn2"><p>Remember that this is a small subset.<a href="#fnref2" class="footnote-back">↩</a></p></li>
<li id="fn3"><p>See the help page of <code>combineSources</code><a href="#fnref3" class="footnote-back">↩</a></p></li>
<li id="fn4"><p>This preparation has been adapted from a <a href="https://support.bioconductor.org/p/48138/#48236">previous discussion</a>.<a href="#fnref4" class="footnote-back">↩</a></p></li>
</ol>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">

        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#introduction">Introduction</a></li>
      <li><a href="#merging">Merging similarities</a></li>
      <li>
<a href="#assessing-a-differential-study">Assessing a differential study</a><ul class="nav nav-pills nav-stacked">
<li><a href="#obtaining-data">Obtaining data</a></li>
      <li><a href="#selecting-differentially-expressed-genes">Selecting differentially expressed genes</a></li>
      <li><a href="#are-differentially-expressed-genes-selected-by-their-functionality">Are differentially expressed genes selected by their functionality?</a></li>
      <li><a href="#are-functionally-related-the-selected-differentially-expressed-genes">Are functionally related the selected differentially expressed genes?</a></li>
      <li><a href="#influence-of-the-fold-change-in-the-functionally-similarity-of-the-genes">Influence of the fold change in the functionally similarity of the genes</a></li>
      </ul>
</li>
      <li><a href="#assessing-a-new-pathway">Assessing a new pathway</a></li>
      <li><a href="#merging-sources-of-information">Merging sources of information</a></li>
      <li><a href="#mirna-analysis">miRNA analysis</a></li>
      <li><a href="#comparing-with-go-similarities">Comparing with GO similarities</a></li>
      <li><a href="#session-info">Session Info</a></li>
      </ul>
</div>
      </div>

</div>



      <footer><div class="copyright">
  <p>Developed by Lluís Revilla Sancho.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.4.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
